<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marnin Wolfe" />


<title>Genomic predictions</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #232629;
    color: #7a7c7d;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #7a7c7d;  padding-left: 4px; }
div.sourceCode
  { color: #cfcfc2; background-color: #232629; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #cfcfc2; } /* Normal */
code span.al { color: #95da4c; background-color: #4d1f24; font-weight: bold; } /* Alert */
code span.an { color: #3f8058; } /* Annotation */
code span.at { color: #2980b9; } /* Attribute */
code span.bn { color: #f67400; } /* BaseN */
code span.bu { color: #7f8c8d; } /* BuiltIn */
code span.cf { color: #fdbc4b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #3daee9; } /* Char */
code span.cn { color: #27aeae; font-weight: bold; } /* Constant */
code span.co { color: #7a7c7d; } /* Comment */
code span.cv { color: #7f8c8d; } /* CommentVar */
code span.do { color: #a43340; } /* Documentation */
code span.dt { color: #2980b9; } /* DataType */
code span.dv { color: #f67400; } /* DecVal */
code span.er { color: #da4453; text-decoration: underline; } /* Error */
code span.ex { color: #0099ff; font-weight: bold; } /* Extension */
code span.fl { color: #f67400; } /* Float */
code span.fu { color: #8e44ad; } /* Function */
code span.im { color: #27ae60; } /* Import */
code span.in { color: #c45b00; } /* Information */
code span.kw { color: #cfcfc2; font-weight: bold; } /* Keyword */
code span.op { color: #cfcfc2; } /* Operator */
code span.ot { color: #27ae60; } /* Other */
code span.pp { color: #27ae60; } /* Preprocessor */
code span.re { color: #2980b9; background-color: #153042; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #da4453; } /* SpecialString */
code span.st { color: #f44f4f; } /* String */
code span.va { color: #27aeae; } /* Variable */
code span.vs { color: #da4453; } /* VerbatimString */
code span.wa { color: #da4453; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">IITA_2021GS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Imputation
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="convertDCas21_6038_ToVCF.html">Conversion DArT report DCas21_6038 to VCF</a>
    </li>
    <li>
      <a href="ImputeDCas21_6038.html">Impute DCas21_6038</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Preliminary Steps
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-cleanTPdata.html">Data Cleaning</a>
    </li>
    <li>
      <a href="02-GetBLUPs.html">Get BLUPs</a>
    </li>
    <li>
      <a href="03-validatePedigree.html">Verify pedigree records</a>
    </li>
    <li>
      <a href="04-PreprocessDataFiles.html">Preprocessing input data</a>
    </li>
  </ul>
</li>
<li>
  <a href="05-CrossValidation.html">Cross-validation</a>
</li>
<li>
  <a href="06-GenomicPredictions.html">Genomic (mate) predictions</a>
</li>
<li>
  <a href="07-Results.html">Results</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Inputs for Sims
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="inputsForSimulationV2.html">Version 2</a>
    </li>
    <li>
      <a href="inputsForSimulation.html">Version 1</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Other
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="about.html">About</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wolfemd/IITA_2021GS">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Genomic predictions</h1>
<h4 class="author">Marnin Wolfe</h4>
<h4 class="date">2021-July-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-18
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>IITA_2021GS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210504code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210504)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210504code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210504)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomwolfemdIITA2021GStree01fb4d2b4934f23be4f3de4308d0fcd825f6e56ctargetblank01fb4d2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/wolfemd/IITA_2021GS/tree/01fb4d2b4934f23be4f3de4308d0fcd825f6e56c" target="_blank">01fb4d2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomwolfemdIITA2021GStree01fb4d2b4934f23be4f3de4308d0fcd825f6e56ctargetblank01fb4d2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/wolfemd/IITA_2021GS/tree/01fb4d2b4934f23be4f3de4308d0fcd825f6e56c" target="_blank">01fb4d2</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/figure/
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  data/DatabaseDownload_2021Aug08/
    Untracked:  data/DatabaseDownload_2021May04/
    Untracked:  data/GBSdataMasterList_31818.csv
    Untracked:  data/IITA_GBStoPhenoMaster_33018.csv
    Untracked:  data/Mate_selection_parental.pool.xlsx
    Untracked:  data/NRCRI_GBStoPhenoMaster_40318.csv
    Untracked:  data/PedigreeGeneticGainCycleTime_aafolabi_01122020.xls
    Untracked:  data/Report-DCas21-6038/
    Untracked:  data/blups_forGP.rds
    Untracked:  data/chr1_RefPanelAndGSprogeny_ReadyForGP_72719.fam
    Untracked:  data/dosages_IITA_2021Aug09.rds
    Untracked:  data/haps_IITA_2021Aug09.rds
    Untracked:  data/recombFreqMat_1minus2c_2021Aug02.qs
    Untracked:  output/

Unstaged changes:
    Modified:   analysis/inputsForSimulationV2.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/06-GenomicPredictions.Rmd</code>) and HTML (<code>docs/06-GenomicPredictions.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/01fb4d2b4934f23be4f3de4308d0fcd825f6e56c/analysis/06-GenomicPredictions.Rmd" target="_blank">01fb4d2</a>
</td>
<td>
wolfemd
</td>
<td>
2021-09-18
</td>
<td>
Updated mate predictions and results page - predicted &gt;250K crosses of 719 parents at Ubiaja now.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/ceae21acfee3b256f2fea1675aa0c690a6fef51b/docs/06-GenomicPredictions.html" target="_blank">ceae21a</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/19c3a3886854b0f6c3ec4ff20e479e9ec6aad70e/docs/06-GenomicPredictions.html" target="_blank">19c3a38</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/e029efce61084b2a8270f54742e9daf9330d9366/docs/06-GenomicPredictions.html" target="_blank">e029efc</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/efebeaba8ea90c30c66b26d4187dcf979711f05a/analysis/06-GenomicPredictions.Rmd" target="_blank">efebeab</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-12
</td>
<td>
Cross-validation and genomic mate predictions complete. All results updated.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/1c03315dbe60717a6f021e182f0b1bf5d054bc65/docs/06-GenomicPredictions.html" target="_blank">1c03315</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/e4df79f49f8c6e7bd1d5c33df830bbfdd4419df9/analysis/06-GenomicPredictions.Rmd" target="_blank">e4df79f</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-11
</td>
<td>
Completed IITA_2021GS pipeline including imputation and genomic prediction. Last bit of cross-validation and cross-prediction finishes in 24 hrs.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/1d017cb48fe22ee9cc976663d199dfaa28100afa/analysis/06-GenomicPredictions.Rmd" target="_blank">1d017cb</a>
</td>
<td>
wolfemd
</td>
<td>
2021-07-25
</td>
<td>
Debugging completed for predictCrosses(). Work shown and tests passed also shown. Full run underway.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/934141c48eff754aa674925be5357d0a2b8be7d3/docs/06-GenomicPredictions.html" target="_blank">934141c</a>
</td>
<td>
wolfemd
</td>
<td>
2021-07-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/cc1eb4b5bb7f38ca04846562d27a3bd4951c6c09/docs/06-GenomicPredictions.html" target="_blank">cc1eb4b</a>
</td>
<td>
wolfemd
</td>
<td>
2021-07-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/772750a8e35e6472113d69e6bc1eff0101aa137f/analysis/06-GenomicPredictions.Rmd" target="_blank">772750a</a>
</td>
<td>
wolfemd
</td>
<td>
2021-07-14
</td>
<td>
DirDom model and selection index calc fully integrated functions.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/b43ee90a8080d23c85ac2721d74bf7ee91f72b4b/analysis/06-GenomicPredictions.Rmd" target="_blank">b43ee90</a>
</td>
<td>
wolfemd
</td>
<td>
2021-07-12
</td>
<td>
Completed predictCrosses function, which predicts cross usefullness on SELINDs and component traits. Work is shown.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/0ac841c57f8403a62167048b7aa9e8263b8ad520/analysis/06-GenomicPredictions.Rmd" target="_blank">0ac841c</a>
</td>
<td>
wolfemd
</td>
<td>
2021-07-11
</td>
<td>
Full genomic prediction including (1) genomic prediction of clone GBLUPs and (2) prediction of cross usefulness. (1) is completed. Work shown. (2) is TO DO still.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="previous-step" class="section level1">
<h1>Previous step</h1>
<ol start="5" style="list-style-type: decimal">
<li><a href="05-CrossValidation.html">Parent-wise cross-validation</a>: Compute parent-wise cross-validation folds using the validated pedigree. Fit models to get marker effects and make subsequent predictions of cross means and (co)variances.</li>
</ol>
</div>
<div id="current-steps" class="section level1">
<h1>Current steps</h1>
<ol style="list-style-type: decimal">
<li><p><strong>Genomic prediction of clone GEBV/GETGV.</strong> Fit GBLUP model, using genotypic add-dom partition. NEW: modelType=“DirDom”, include genome-wide inbreeding effect in GEBV/GETGV predictions <em>after</em> backsolving SNP effects. For all models, extract GBLUPs and backsolve SNP effects for use in cross usefulness predictions (mean+variance predictions). <strong>ALSO NEW</strong>: selection index predictions.</p></li>
<li><p><strong>Genomic prediction of cross</strong> <span class="math inline">\(UC_{parent}\)</span> and <span class="math inline">\(UC_{variety}\)</span>. Rank potential parents on SI. Predict all possible crosses of some portion of best parents.</p></li>
</ol>
</div>
<div id="genomic-prediction-of-clone-gebvgetgv" class="section level1">
<h1>Genomic prediction of clone GEBV/GETGV</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1) start a screen shell </span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">screen</span><span class="kw">;</span> <span class="co"># or screen -r if re-attaching...</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 2) start the singularity Linux shell inside that</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="ex">singularity</span> shell ~/rocker2.sif<span class="kw">;</span> </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Project directory, so R will use as working dir.</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /home/mw489/IITA_2021GS/<span class="kw">;</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 3) Start R</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="ex">R</span></span></code></pre></div>
<p>Load input data</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># NEEDED LIBRARIES</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse); <span class="fu">require</span>(magrittr); <span class="fu">library</span>(qs)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genomicMateSelectR)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># BLUPs</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>blups<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;blups_forGP.rds&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>varcomp) <span class="sc">%&gt;%</span> </span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">TrainingData=</span>blups) <span class="co"># for compatibility with runCrossVal() function </span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># DOSAGE MATRIX (UNFILTERED)</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>dosages<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;dosages_IITA_2021Aug09.rds&quot;</span>))</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="co"># SNP SETS TO ANALYZE</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>snpsets<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;snpsets.rds&quot;</span>))</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co"># SELECTION INDEX WEIGHTS</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="do">## from IYR+IK</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="do">## note that not ALL predicted traits are on index</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>SIwts<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="at">logFYLD=</span><span class="dv">20</span>,</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">HI=</span><span class="dv">10</span>,</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">DM=</span><span class="dv">15</span>,</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">MCMDS=</span><span class="sc">-</span><span class="dv">10</span>,</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">logRTNO=</span><span class="dv">12</span>,</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">logDYLD=</span><span class="dv">20</span>,</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">logTOPYLD=</span><span class="dv">15</span>,</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">PLTHT=</span><span class="dv">10</span>) </span></code></pre></div>
<p>Run the “DirDom” modelTypes built into <code>runGenomicPredictions()</code>. Output will contain both GBLUPs for selection of clones <em>and</em> SNP effects to use as input for prediction of cross usefulness and subsequent mate selection.</p>
<p>Get effects for: <strong>full_set</strong> (~31K SNPs), <strong>medium_set</strong> (~13K, LD-pruned) and <strong>reduced_set</strong> (~9K SNPs, LD-pruned).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cbsulm30 - 112 cores, 512 GB RAM - 2021 Aug 10 - 8:40am </span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>snpSet<span class="ot">&lt;-</span><span class="st">&quot;full_set&quot;</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>grms<span class="ot">&lt;-</span><span class="fu">list</span>(<span class="at">A=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_A_IITA_2021Aug09.rds&quot;</span>)), </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">D=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_Dgeno_IITA_2021Aug09.rds&quot;</span>)))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;Time elapsed: 24.066 mins&quot;</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># cbsulm30 - 112 cores, 512 GB RAM - 2021 Aug 10 - 8:40am</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>snpSet<span class="ot">&lt;-</span><span class="st">&quot;reduced_set&quot;</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>grms<span class="ot">&lt;-</span><span class="fu">list</span>(<span class="at">A=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_A_ReducedSNPset_IITA_2021Aug09.rds&quot;</span>)), </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>           <span class="at">D=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_Dgeno_ReducedSNPset_IITA_2021Aug09.rds&quot;</span>)))</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;Time elapsed: 22.431 mins&quot;</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co"># cbsulm20 - 88 cores, 512 GB RAM - 2021 Aug 10 - 2:50pm</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>snpSet<span class="ot">&lt;-</span><span class="st">&quot;medium_set&quot;</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>grms<span class="ot">&lt;-</span><span class="fu">list</span>(<span class="at">A=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_A_MediumSNPset_IITA_2021Aug09.rds&quot;</span>)), </span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>           <span class="at">D=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_Dgeno_MediumSNPset_IITA_2021Aug09.rds&quot;</span>)))</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;Time elapsed: 13.299 mins&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>snps2keep<span class="ot">&lt;-</span>snpsets <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Set<span class="sc">==</span>snpSet) <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(snps2keep) <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(snps2keep)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>dosages<span class="ot">&lt;-</span>dosages[,snps2keep<span class="sc">$</span>FULL_SNP_ID]</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(snpsets); <span class="fu">gc</span>()</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>starttime<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>gpreds<span class="ot">&lt;-</span><span class="fu">runGenomicPredictions</span>(<span class="at">modelType=</span><span class="st">&quot;DirDom&quot;</span>,<span class="at">selInd=</span><span class="cn">TRUE</span>, <span class="at">SIwts=</span>SIwts,</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>                              <span class="at">getMarkEffs=</span><span class="cn">TRUE</span>,</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>                              <span class="at">returnPEV=</span><span class="cn">FALSE</span>,</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>                              <span class="at">blups=</span>blups,<span class="at">grms=</span>grms,<span class="at">dosages=</span>dosages,</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>                              <span class="at">ncores=</span><span class="dv">11</span>,<span class="at">nBLASthreads=</span><span class="dv">5</span>)</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(gpreds,</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">paste0</span>(<span class="st">&quot;genomicPredictions_&quot;</span>,snpSet,<span class="st">&quot;_2021Aug09.rds&quot;</span>)))</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>endtime<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]; <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">&quot;Time elapsed: &quot;</span>,</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">round</span>((endtime<span class="sc">-</span>starttime)<span class="sc">/</span><span class="dv">60</span>,<span class="dv">3</span>),<span class="st">&quot; mins&quot;</span>))</span></code></pre></div>
</div>
<div id="genomic-prediction-of-cross-usefulness" class="section level1">
<h1>Genomic prediction of cross usefulness</h1>
<ol style="list-style-type: decimal">
<li>Use GBLUPs on SELIND to choose top fraction of the clones. <strong>Cross-reference with ‘accessions_infield’ list</strong>.</li>
<li>For those selected parents, predict the SELIND usefulness for all pairwise matings.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1) start a screen shell </span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="ex">screen</span><span class="kw">;</span> <span class="co"># or screen -r if re-attaching...</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 2) start the singularity Linux shell inside that</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="ex">singularity</span> shell ~/rocker2.sif<span class="kw">;</span> </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Project directory, so R will use as working dir.</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /home/mw489/IITA_2021GS/<span class="kw">;</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 3) Start R</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="ex">R</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># NEEDED LIBRARIES</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse); <span class="fu">require</span>(magrittr); <span class="fu">library</span>(qs)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genomicMateSelectR)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co"># BLUPs</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>blups<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;blups_forGP.rds&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>varcomp)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co"># DOSAGE MATRIX (UNFILTERED)</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>dosages<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;dosages_IITA_2021Aug09.rds&quot;</span>))</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co"># RECOMBINATION FREQUENCY MATRIX (UNFILTERED)</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>recombFreqMat<span class="ot">&lt;-</span><span class="fu">qread</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">&quot;recombFreqMat_1minus2c_2021Aug02.qs&quot;</span>))</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="co"># HAPLOTYPE MATRIX (UNFILTERED)</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="do">## keep only haplos for parents-in-the-pedigree</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="do">## those which will be used in prediction, saves memory</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>haploMat<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;haps_IITA_2021Aug09.rds&quot;</span>))</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>parents<span class="ot">&lt;-</span><span class="fu">union</span>(ped<span class="sc">$</span>sireID,ped<span class="sc">$</span>damID) </span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>parenthaps<span class="ot">&lt;-</span><span class="fu">sort</span>(<span class="fu">c</span>(<span class="fu">paste0</span>(parents,<span class="st">&quot;_HapA&quot;</span>),</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">paste0</span>(parents,<span class="st">&quot;_HapB&quot;</span>)))</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>haploMat<span class="ot">&lt;-</span>haploMat[parenthaps,]</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a><span class="co"># SNP SETS TO ANALYZE</span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>snpsets<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;snpsets.rds&quot;</span>))</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="co"># SELECTION INDEX WEIGHTS</span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a><span class="do">## from IYR+IK</span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a><span class="do">## note that not ALL predicted traits are on index</span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>SIwts<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="at">logFYLD=</span><span class="dv">20</span>,</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">HI=</span><span class="dv">10</span>,</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>         <span class="at">DM=</span><span class="dv">15</span>,</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">MCMDS=</span><span class="sc">-</span><span class="dv">10</span>,</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">logRTNO=</span><span class="dv">12</span>,</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">logDYLD=</span><span class="dv">20</span>,</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">logTOPYLD=</span><span class="dv">15</span>,</span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">PLTHT=</span><span class="dv">10</span>) </span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a><span class="co"># LIST OF ACCESSIONS LIKELY IN THE FIELD</span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>accessions_infield<span class="ot">&lt;-</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,</span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">&quot;accessions_possibly_infield_2021Aug10.rds&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># gpreds_full&lt;-readRDS(file = here::here(&quot;output&quot;,&quot;genomicPredictions_full_set_2021Aug09.rds&quot;))</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># gpreds_medium&lt;-readRDS(file = here::here(&quot;output&quot;,&quot;genomicPredictions_medium_set_2021Aug09.rds&quot;))</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># gpreds_reduced&lt;-readRDS(file = here::here(&quot;output&quot;,&quot;genomicPredictions_reduced_set_2021Aug09.rds&quot;))</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="do">### Quick check that GBLUPs from full and medium_set are strongly correlated</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="do">### This will be an additional assurance that similar cross variances</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="do">### will be predicted by the reduced SNP model</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="do">### Cor between SELIND GEBV and GETGV between full_set and reduced_set of SNPs?</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># left_join(gpreds_full$gblups[[1]] %&gt;% select(GID,predOf,SELIND) %&gt;% rename(SELIND_full=SELIND),</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co">#           gpreds_reduced$gblups[[1]] %&gt;% select(GID,predOf,SELIND) %&gt;% rename(SELIND_reduced=SELIND)) %&gt;%</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(predOf) %&gt;% </span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   summarize(SELIND_corModels=cor(SELIND_full,SELIND_reduced))</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co"># predOf  SELIND_corModels</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="co"># GEBV  0.0363334           </span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co"># GETGV 0.7809016   </span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="do">## TERRIBLE!! :(</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="do">### Cor between SELIND GEBV and GETGV between full_set and medium_set of SNPs?</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="co"># left_join(gpreds_full$gblups[[1]] %&gt;% select(GID,predOf,SELIND) %&gt;% rename(SELIND_full=SELIND),</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="co">#           gpreds_medium$gblups[[1]] %&gt;% select(GID,predOf,SELIND) %&gt;% rename(SELIND_medium=SELIND)) %&gt;%</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(predOf) %&gt;% </span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="co">#   summarize(SELIND_corModels=cor(SELIND_full,SELIND_medium))</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co"># predOf  SELIND_corModels</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="co"># GEBV  0.9901325           </span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a><span class="co"># GETGV 0.9887326</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="do">## EXCELLENT!! :)</span></span></code></pre></div>
<p>Choose the best parents for which to predict crosses. Use the GBLUPs from the <strong>full_set</strong> of SNPs.</p>
<p>Take the union of the top 300 clones on the SELIND in terms of GEBV and of GETGV. Probably they will be a very similar list.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># SELECT THE BEST PARENTS AS CROSSES-TO-BE-PREDICTED</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>nParentsToSelect<span class="ot">&lt;-</span><span class="dv">300</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>gpreds_full<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicPredictions_full_set_2021Aug09.rds&quot;</span>))</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>union_bestGEBVandGETGV<span class="ot">&lt;-</span><span class="fu">union</span>(gpreds_full<span class="sc">$</span>gblups[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">filter</span>(predOf<span class="sc">==</span><span class="st">&quot;GEBV&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">arrange</span>(<span class="fu">desc</span>(SELIND)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span>nParentsToSelect) <span class="sc">%$%</span> GID,</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                              gpreds_full<span class="sc">$</span>gblups[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">filter</span>(predOf<span class="sc">==</span><span class="st">&quot;GETGV&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">arrange</span>(<span class="fu">desc</span>(SELIND)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span>nParentsToSelect) <span class="sc">%$%</span> GID)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(gpreds_full); </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(union_bestGEBVandGETGV) </span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] 365 parents in top nParentsToSelect on SELIND for GEBV/GETGV </span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="co"># KEEP ONLY CANDIDATE PARENTS EXPECTED TO BE IN THE FIELD</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(union_bestGEBVandGETGV <span class="sc">%in%</span> accessions_infield<span class="sc">$</span>FullSampleName)</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="co"># FALSE  TRUE </span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   165   200</span></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>parentsToPredictCrosses<span class="ot">&lt;-</span>union_bestGEBVandGETGV <span class="sc">%&gt;%</span> </span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>  .[. <span class="sc">%in%</span> accessions_infield<span class="sc">$</span>FullSampleName]</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>CrossesToPredict<span class="ot">&lt;-</span><span class="fu">crosses2predict</span>(parentsToPredictCrosses)</span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(CrossesToPredict)</span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] 20100 possible crosses of 200 parents</span></span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(parentsToPredictCrosses,</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>                          <span class="st">&quot;parentsToPredictCrosses_2021Aug10.rds&quot;</span>))</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(CrossesToPredict,</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>                          <span class="st">&quot;CrossesToPredict_2021Aug10.rds&quot;</span>))</span></code></pre></div>
<p>Predict all pairwise crosses of those 200 parents.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cbsulm17 - 112 cores, 512 GB RAM - 2021 Aug 11 - 8:10am</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>snpSet<span class="ot">&lt;-</span><span class="st">&quot;full_set&quot;</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>grms<span class="ot">&lt;-</span><span class="fu">list</span>(<span class="at">A=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_A_IITA_2021Aug09.rds&quot;</span>)), </span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">D=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_Dgeno_IITA_2021Aug09.rds&quot;</span>)))</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>predTheMeans<span class="ot">&lt;-</span><span class="cn">TRUE</span>; predTheVars<span class="ot">&lt;-</span><span class="cn">FALSE</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>gpreds<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicPredictions_full_set_2021Aug09.rds&quot;</span>)); <span class="fu">gc</span>()</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="do">## takes ~2 minutes, predicting means only</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co"># cbsulm17 - 112 cores, 512 GB RAM - 2021 Aug 10 - 8:25am</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>snpSet<span class="ot">&lt;-</span><span class="st">&quot;medium_set&quot;</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>grms<span class="ot">&lt;-</span><span class="fu">list</span>(<span class="at">A=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_A_MediumSNPset_IITA_2021Aug09.rds&quot;</span>)), </span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>           <span class="at">D=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_Dgeno_MediumSNPset_IITA_2021Aug09.rds&quot;</span>)))</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>predTheMeans<span class="ot">&lt;-</span><span class="cn">TRUE</span>; predTheVars<span class="ot">&lt;-</span><span class="cn">TRUE</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>gpreds<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicPredictions_medium_set_2021Aug09.rds&quot;</span>)); <span class="fu">gc</span>()</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="co"># 1263.904</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>snps2keep<span class="ot">&lt;-</span>snpsets <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Set<span class="sc">==</span>snpSet) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(snps2keep) <span class="sc">%&gt;%</span> </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(snps2keep)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>dosages<span class="ot">&lt;-</span>dosages[,snps2keep<span class="sc">$</span>FULL_SNP_ID]</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>haploMat<span class="ot">&lt;-</span>haploMat[,snps2keep<span class="sc">$</span>FULL_SNP_ID]</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>recombFreqMat<span class="ot">&lt;-</span>recombFreqMat[snps2keep<span class="sc">$</span>FULL_SNP_ID,snps2keep<span class="sc">$</span>FULL_SNP_ID]</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(snpsets); <span class="fu">gc</span>()</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>snpSet; <span class="fu">dim</span>(dosages); predTheMeans; predTheVars</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>start<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>crossPreds<span class="ot">&lt;-</span><span class="fu">predictCrosses</span>(<span class="at">modelType=</span><span class="st">&quot;DirDom&quot;</span>,<span class="at">stdSelInt =</span> <span class="fl">2.0627128</span>, </span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>                           <span class="at">selInd=</span><span class="cn">TRUE</span>, <span class="at">SIwts=</span>SIwts,</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>                           <span class="at">CrossesToPredict=</span>CrossesToPredict,</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>                           <span class="at">snpeffs=</span>gpreds<span class="sc">$</span>genomicPredOut[[<span class="dv">1</span>]],</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>                           <span class="at">dosages=</span>dosages,</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>                           <span class="at">haploMat=</span>haploMat,<span class="at">recombFreqMat=</span>recombFreqMat,</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>                           <span class="at">ncores=</span><span class="dv">20</span>,<span class="at">nBLASthreads=</span><span class="dv">5</span>,</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>                           <span class="at">predTheMeans =</span> predTheMeans,</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>                           <span class="at">predTheVars =</span> predTheVars)</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>runtime<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]<span class="sc">-</span>start; runtime<span class="sc">/</span><span class="dv">60</span></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(crossPreds,<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>                                     <span class="fu">paste0</span>(<span class="st">&quot;genomicMatePredictions_&quot;</span>,</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>                                            snpSet,<span class="st">&quot;_2021Aug10.rds&quot;</span>)))</span></code></pre></div>
<div id="predict-crosses-for-clones-in-ubiaja" class="section level2">
<h2>Predict crosses for clones in Ubiaja</h2>
<p>September 10th, 2021: received from I. Kayondo a list of plots in Ubiaja available for crosses (<code>data/Mate_selection_parental.pool.xlsx</code>). Match these lines to genomic data and (if not already done) predict matings!</p>
<p>First thing, match the germplasm in the list provided to genotype data…</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genomicMateSelectR)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>dbdata<span class="ot">&lt;-</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;IITA_ExptDesignsDetected_2021Aug08.rds&quot;</span>))</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>dosages<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;dosages_IITA_2021Aug09.rds&quot;</span>))</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>parent_candidates<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;Mate_selection_parental.pool.xlsx&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(accession_name) <span class="sc">%&gt;%</span> </span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">germplasmName=</span>accession_name) <span class="sc">%&gt;%</span> </span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(dbdata <span class="sc">%&gt;%</span> </span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>              <span class="fu">distinct</span>(germplasmName,GID,FullSampleName))</span></code></pre></div>
<p>1354 unique germplasm to select among. Some genotyped multiple times, so 1576 geno matches.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>parent_candidates <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(GID)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(germplasmName) <span class="sc">%&gt;%</span> </span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(.,<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;parent_candidates_NOGENOS.txt&quot;</span>),<span class="at">col.names =</span> F, <span class="at">row.names =</span> F, <span class="at">quote =</span> F)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>parent_candidates <span class="sc">%&gt;%</span> </span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>FullSampleName <span class="sc">%in%</span> <span class="fu">rownames</span>(dosages),</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>         <span class="sc">!</span><span class="fu">is.na</span>(FullSampleName)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(germplasmName) <span class="sc">%&gt;%</span> </span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(.,<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;parent_candidates_PEDIGREEREJECT.txt&quot;</span>),<span class="at">col.names =</span> F, <span class="at">row.names =</span> F, <span class="at">quote =</span> F)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>parent_candidates <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FullSampleName <span class="sc">%in%</span> <span class="fu">rownames</span>(dosages)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(germplasmName,FullSampleName) <span class="sc">%&gt;%</span> </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(germplasmName) <span class="sc">%&gt;%</span> </span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span>)</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>parent_candidates <span class="sc">%&gt;%</span> </span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(.,<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;parent_candidates_TOPREDICT.txt&quot;</span>),<span class="at">col.names =</span> F, <span class="at">row.names =</span> F, <span class="at">quote =</span> F)</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>CrossesToPredict<span class="ot">&lt;-</span>genomicMateSelectR<span class="sc">::</span><span class="fu">crosses2predict</span>(parents)</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a><span class="co"># nrow(CrossesToPredict)</span></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] 258840 possible crosses of 719 parents</span></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(CrossesToPredict,</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>                          <span class="st">&quot;CrossesToPredict_2021Sep13.rds&quot;</span>))</span></code></pre></div>
<p>13 lines don’t seem to have matches in SNP data. 552 lines DO have SNP data, but seem to have been excluded from my analyses at the <a href="03-validatePedigree.html">pedigree-validation step here</a> because they had neither BLUPs (useful training data) nor could their pedigree be validated.</p>
<p>That might not be what you want, since as long as the genomic data matches the clone in the field, even without a verified pedigree it could be a useful parent for crossing. Also, my pedigree verification could have false rejections. So care should be taken. Two lists of clones for you: <code>output/parent_candidates_NOGENOS.txt</code>, <code>output/parent_candidates_PEDIGREEREJECT.txt</code>.</p>
<p>719 germplasmName have genotype data matches and verified pedigrees. Some have more than one DNA sample, so I will choose one for each.</p>
<p>Because there are PLENTY of candidate parents, I suggest to proceed for now, but some action or decision should be made about the 552 germplasm not-yet-considered.</p>
<p>File with the list I suggeset to work from: <code>ouptput/parent_candidates_TOPREDICT.txt</code>.</p>
<p>You can match the predictions I make with your field plots based on match the “germplasmName” column in this file with the “accession_name” column in the <code>Mate_selection_parental.pool.xlsx</code> file you shared with me.</p>
<p>Regarding the number of crosses I can predict: It took 21 hrs on a single 112 core server to predict 20100 crosses (pairwise for 200 parents).</p>
<p>For pairwise crosses among 719 in-field parents, 258840 crosses-to-predict. 271 hrs or 11 days. I could secure at least 3 servers and do it in about 3-5 actual days.</p>
<p>Or I could take the top 500 parents (125250 crosses) or 400 parents (80200 crosses) and cut it down to as little as 2 days.</p>
<p>Delivery end of week vs mid-week?</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1) start a screen shell </span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="ex">screen</span><span class="kw">;</span> <span class="co"># or screen -r if re-attaching...</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 2) start the singularity Linux shell inside that</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="ex">singularity</span> shell ~/rocker2.sif<span class="kw">;</span> </span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Project directory, so R will use as working dir.</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /home/mw489/IITA_2021GS/<span class="kw">;</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 3) Start R</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="ex">R</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># NEEDED LIBRARIES</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse); <span class="fu">require</span>(magrittr); <span class="fu">library</span>(qs)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genomicMateSelectR)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="co"># BLUPs</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>blups<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;blups_forGP.rds&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>varcomp)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="co"># DOSAGE MATRIX (UNFILTERED)</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>dosages<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;dosages_IITA_2021Aug09.rds&quot;</span>))</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a><span class="co"># RECOMBINATION FREQUENCY MATRIX (UNFILTERED)</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>recombFreqMat<span class="ot">&lt;-</span><span class="fu">qread</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">&quot;recombFreqMat_1minus2c_2021Aug02.qs&quot;</span>))</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="co"># HAPLOTYPE MATRIX (UNFILTERED)</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a><span class="do">## keep only haplos for parents-in-the-pedigree</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a><span class="do">## those which will be used in prediction, saves memory</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>haploMat<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;haps_IITA_2021Aug09.rds&quot;</span>))</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a><span class="co"># PARENTS FOR CROSS PREDICTIONS</span></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>parents<span class="ot">&lt;-</span><span class="fu">read.table</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;parent_candidates_TOPREDICT.txt&quot;</span>), <span class="at">header =</span> F, <span class="at">stringsAsFactors =</span> F)<span class="sc">$</span>V2</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a><span class="co"># SUBSET HAPS</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>parenthaps<span class="ot">&lt;-</span><span class="fu">sort</span>(<span class="fu">c</span>(<span class="fu">paste0</span>(parents,<span class="st">&quot;_HapA&quot;</span>),</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">paste0</span>(parents,<span class="st">&quot;_HapB&quot;</span>)))</span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>haploMat<span class="ot">&lt;-</span>haploMat[parenthaps,]</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a><span class="co"># SNP SETS TO ANALYZE</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>snpsets<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;snpsets.rds&quot;</span>))</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a><span class="co"># SELECTION INDEX WEIGHTS</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a><span class="do">## from IYR+IK</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a><span class="do">## note that not ALL predicted traits are on index</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>SIwts<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="at">logFYLD=</span><span class="dv">20</span>,</span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">HI=</span><span class="dv">10</span>,</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">DM=</span><span class="dv">15</span>,</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">MCMDS=</span><span class="sc">-</span><span class="dv">10</span>,</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">logRTNO=</span><span class="dv">12</span>,</span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>         <span class="at">logDYLD=</span><span class="dv">20</span>,</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">logTOPYLD=</span><span class="dv">15</span>,</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">PLTHT=</span><span class="dv">10</span>) </span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a><span class="co"># CROSSES TO PREDICT</span></span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>CrossesToPredict<span class="ot">&lt;-</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,</span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">&quot;CrossesToPredict_2021Sep13.rds&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cbsulm26 - 88 cores, 512 GB RAM - 2021 Sep 13</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>snpSet<span class="ot">&lt;-</span><span class="st">&quot;full_set&quot;</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>grms<span class="ot">&lt;-</span><span class="fu">list</span>(<span class="at">A=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_A_IITA_2021Aug09.rds&quot;</span>)), </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">D=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_Dgeno_IITA_2021Aug09.rds&quot;</span>)))</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>predTheMeans<span class="ot">&lt;-</span><span class="cn">TRUE</span>; predTheVars<span class="ot">&lt;-</span><span class="cn">FALSE</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>gpreds<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicPredictions_full_set_2021Aug09.rds&quot;</span>)); <span class="fu">gc</span>()</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>snps2keep<span class="ot">&lt;-</span>snpsets <span class="sc">%&gt;%</span> </span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Set<span class="sc">==</span>snpSet) <span class="sc">%&gt;%</span> </span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(snps2keep) <span class="sc">%&gt;%</span> </span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(snps2keep)</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>dosages<span class="ot">&lt;-</span>dosages[,snps2keep<span class="sc">$</span>FULL_SNP_ID]</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>haploMat<span class="ot">&lt;-</span>haploMat[,snps2keep<span class="sc">$</span>FULL_SNP_ID]</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>recombFreqMat<span class="ot">&lt;-</span>recombFreqMat[snps2keep<span class="sc">$</span>FULL_SNP_ID,snps2keep<span class="sc">$</span>FULL_SNP_ID]</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(snpsets); <span class="fu">gc</span>()</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>snpSet; <span class="fu">dim</span>(dosages); predTheMeans; predTheVars</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>start<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>crossPreds<span class="ot">&lt;-</span><span class="fu">predictCrosses</span>(<span class="at">modelType=</span><span class="st">&quot;DirDom&quot;</span>,<span class="at">stdSelInt =</span> <span class="fl">2.0627128</span>, </span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>                           <span class="at">selInd=</span><span class="cn">TRUE</span>, <span class="at">SIwts=</span>SIwts,</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>                           <span class="at">CrossesToPredict=</span>CrossesToPredict,</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>                           <span class="at">snpeffs=</span>gpreds<span class="sc">$</span>genomicPredOut[[<span class="dv">1</span>]],</span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>                           <span class="at">dosages=</span>dosages,</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>                           <span class="at">haploMat=</span>haploMat,<span class="at">recombFreqMat=</span>recombFreqMat,</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>                           <span class="at">ncores=</span><span class="dv">17</span>,<span class="at">nBLASthreads=</span><span class="dv">5</span>,</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>                           <span class="at">predTheMeans =</span> predTheMeans,</span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>                           <span class="at">predTheVars =</span> predTheVars)</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>runtime<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]<span class="sc">-</span>start; runtime<span class="sc">/</span><span class="dv">60</span></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(crossPreds,<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>                                     <span class="fu">paste0</span>(<span class="st">&quot;genomicMatePredictions_&quot;</span>,</span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>                                            snpSet,<span class="st">&quot;_2021Sep13.rds&quot;</span>)))</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a><span class="co"># elapsed </span></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a><span class="co"># 14.7936 minutes</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>snpSet<span class="ot">&lt;-</span><span class="st">&quot;medium_set&quot;</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>grms<span class="ot">&lt;-</span><span class="fu">list</span>(<span class="at">A=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_A_MediumSNPset_IITA_2021Aug09.rds&quot;</span>)), </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">D=</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;kinship_Dgeno_MediumSNPset_IITA_2021Aug09.rds&quot;</span>)))</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>predTheMeans<span class="ot">&lt;-</span><span class="cn">TRUE</span>; predTheVars<span class="ot">&lt;-</span><span class="cn">TRUE</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>gpreds<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicPredictions_medium_set_2021Aug09.rds&quot;</span>)); <span class="fu">gc</span>()</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>snps2keep<span class="ot">&lt;-</span>snpsets <span class="sc">%&gt;%</span> </span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Set<span class="sc">==</span>snpSet) <span class="sc">%&gt;%</span> </span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(snps2keep) <span class="sc">%&gt;%</span> </span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(snps2keep)</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>dosages<span class="ot">&lt;-</span>dosages[,snps2keep<span class="sc">$</span>FULL_SNP_ID]</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>haploMat<span class="ot">&lt;-</span>haploMat[,snps2keep<span class="sc">$</span>FULL_SNP_ID]</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>recombFreqMat<span class="ot">&lt;-</span>recombFreqMat[snps2keep<span class="sc">$</span>FULL_SNP_ID,snps2keep<span class="sc">$</span>FULL_SNP_ID]</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(snpsets); <span class="fu">gc</span>()</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>snpSet; <span class="fu">dim</span>(dosages); predTheMeans; predTheVars</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="co"># DIVIDE CROSSES-TO-PREDICT INTO 3 CHUNKS</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>CrossesToPredict <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Chunk=</span><span class="fu">rep_along</span>(<span class="at">along =</span> CrossesToPredict<span class="sc">$</span>sireID, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>) <span class="sc">%&gt;%</span> <span class="fu">sort</span>(.)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nest</span>(<span class="at">Crosses=</span><span class="fu">c</span>(sireID,damID))</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a><span class="co"># cbsulm26 - 88 cores, 512 GB RAM - 2021 Sep 13, 10:20am</span></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>CrossesToPredict<span class="ot">&lt;-</span>CrossesToPredict<span class="sc">$</span>Crosses[[<span class="dv">1</span>]]</span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>chunk<span class="ot">&lt;-</span><span class="dv">1</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;Done predicting fam vars. Took 3867.09 mins for 86280 crosses&quot; = 64.45 hrs = 2.68 days</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;Done predicting fam vars. Took 1658.5 mins for 86280 crosses&quot;</span></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a><span class="co"># 5548.574</span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a><span class="do">## two runs of predCrossVars() function internally </span></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a><span class="do">## to predictCrosses() for the DirDom model</span></span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a><span class="do">## 3867.09/86280 = 0.045 min per cross</span></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a><span class="do">## (3867.09*2)/86280 = 0.089 total mins per cross (for DirDom model)</span></span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a><span class="co"># cbsulm18 - 88 cores, 512 GB RAM - 2021 Sep 13, 9:30am</span></span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>CrossesToPredict<span class="ot">&lt;-</span>CrossesToPredict<span class="sc">$</span>Crosses[[<span class="dv">2</span>]]</span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>chunk<span class="ot">&lt;-</span><span class="dv">2</span></span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;Done predicting fam vars. Took 3700.52 mins for 86280 crosses&quot; = 61.67 hrs = 2.57 days</span></span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;Done predicting fam vars. Took 1600.63 mins for 86280 crosses&quot;</span></span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a><span class="co"># 5323.687 mins elapsed total</span></span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a><span class="co"># cbsulm19 - 88 cores, 512 GB RAM - 2021 Sep 13, 9:30am</span></span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>CrossesToPredict<span class="ot">&lt;-</span>CrossesToPredict<span class="sc">$</span>Crosses[[<span class="dv">3</span>]]</span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>chunk<span class="ot">&lt;-</span><span class="dv">3</span></span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;Done predicting fam vars. Took 3659.66 mins for 86280 crosses&quot;</span></span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;Done predicting fam vars. Took 1609.88 mins for 86280 crosses&quot;</span></span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a><span class="co"># 5292.067 = 88.2 hrs = 3.67 days</span></span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>CrossesToPredict <span class="sc">%&gt;%</span> head</span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a><span class="do">## 128.9 hrs = 5.37 days expected for 3 servers (each 88 core, 512 GB RAM)</span></span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a><span class="do">## to deliver ~260K cross predictions for an directional dominance model with</span></span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a><span class="do">## an 8 trait SI and 13K SNPs</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>start<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>crossPreds<span class="ot">&lt;-</span><span class="fu">predictCrosses</span>(<span class="at">modelType=</span><span class="st">&quot;DirDom&quot;</span>,<span class="at">stdSelInt =</span> <span class="fl">2.0627128</span>, </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">selInd=</span><span class="cn">TRUE</span>, <span class="at">SIwts=</span>SIwts,</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">CrossesToPredict=</span>CrossesToPredict,</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">snpeffs=</span>gpreds<span class="sc">$</span>genomicPredOut[[<span class="dv">1</span>]],</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">dosages=</span>dosages,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">haploMat=</span>haploMat,<span class="at">recombFreqMat=</span>recombFreqMat,</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">ncores=</span><span class="dv">17</span>,<span class="at">nBLASthreads=</span><span class="dv">5</span>,</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">predTheMeans =</span> predTheMeans,</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">predTheVars =</span> predTheVars)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>runtime<span class="ot">&lt;-</span><span class="fu">proc.time</span>()[<span class="dv">3</span>]<span class="sc">-</span>start; runtime<span class="sc">/</span><span class="dv">60</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(crossPreds,<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>                                     <span class="fu">paste0</span>(<span class="st">&quot;genomicMatePredictions_&quot;</span>,</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>                                            snpSet,<span class="st">&quot;_chunk&quot;</span>,chunk,<span class="st">&quot;_2021Sep13.rds&quot;</span>)))</span></code></pre></div>
</div>
</div>
<div id="write-tidy-breeder-friendly-predictions-to-disk" class="section level1">
<h1>Write tidy breeder-friendly predictions to disk</h1>
<div id="august-2021" class="section level2">
<h2>August 2021</h2>
<p>Add genetic groups and cohort identifiers and tidy format</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>gpreds<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicPredictions_full_set_2021Aug09.rds&quot;</span>)) </span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>gpreds<span class="sc">$</span>gblups[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GeneticGroup=</span><span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;C1&quot;</span>,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">grepl</span>(<span class="st">&quot;TMS14&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;C2&quot;</span>,</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">grepl</span>(<span class="st">&quot;TMS15&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;C3&quot;</span>,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">grepl</span>(<span class="st">&quot;TMS18&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;C4&quot;</span>,</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">grepl</span>(<span class="st">&quot;TMS20&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;C5&quot;</span>,</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">grepl</span>(<span class="st">&quot;TMS16&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;TMS16&quot;</span>,</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">grepl</span>(<span class="st">&quot;TMS17&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;TMS17&quot;</span>,</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">grepl</span>(<span class="st">&quot;TMS19&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;TMS19&quot;</span>,</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>                                <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13|TMS14|TMS15|TMS16|TMS17|TMS18|TMS19|TMS20&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;PreGS&quot;</span>),</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>         <span class="co">#GeneticGroup=factor(GeneticGroup,levels = c(&quot;PreGS&quot;,&quot;C1&quot;,&quot;C2&quot;,&quot;C3&quot;,&quot;C4&quot;,&quot;C5&quot;,&quot;TMS16&quot;,&quot;TMS17&quot;,&quot;TMS19&quot;)),</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">Cohort=</span><span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;TMS13&quot;</span>,</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">grepl</span>(<span class="st">&quot;TMS14&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;TMS14&quot;</span>,</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">grepl</span>(<span class="st">&quot;TMS15&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;TMS15&quot;</span>,</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">grepl</span>(<span class="st">&quot;TMS16&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;TMS16&quot;</span>,</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">grepl</span>(<span class="st">&quot;TMS17&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;TMS17&quot;</span>,</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">grepl</span>(<span class="st">&quot;TMS18&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;TMS18&quot;</span>,</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">grepl</span>(<span class="st">&quot;TMS19&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;TMS19&quot;</span>,</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">grepl</span>(<span class="st">&quot;TMS20&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;TMS20&quot;</span>,</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>                          <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13|TMS14|TMS15|TMS16|TMS17|TMS18|TMS19|TMS20&quot;</span>,GID)<span class="sc">~</span><span class="st">&quot;PreGS&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(GeneticGroup,<span class="at">.after =</span> <span class="st">&quot;predOf&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(Cohort,<span class="at">.after =</span> <span class="st">&quot;GeneticGroup&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(predOf,<span class="fu">desc</span>(SELIND)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.csv</span>(.,<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicPredictions_full_set_2021Aug09.csv&quot;</span>),</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> F)</span></code></pre></div>
<p><strong>NOTE:</strong> For cross predictions, check that the predMean from full and medium set are highly correlated. As long as that is true, combine the predMean from full set with pred var from medium set.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>crossPreds_full<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicMatePredictions_full_set_2021Aug10.rds&quot;</span>))</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>crossPreds_medium<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicMatePredictions_medium_set_2021Aug10.rds&quot;</span>))</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>crossPreds_medium<span class="sc">$</span>rawPreds[[<span class="dv">1</span>]]</span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>crossPreds_full<span class="sc">$</span>tidyPreds[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">predMean_full=</span>predMean) <span class="sc">%&gt;%</span> </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(crossPreds_medium<span class="sc">$</span>tidyPreds[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rename</span>(<span class="at">predMean_medium=</span>predMean)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(predOf,Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corPredMeans=</span><span class="fu">cor</span>(predMean_full,predMean_medium),<span class="at">.groups =</span> <span class="st">&#39;drop&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(corPredMeans)) <span class="sc">%$%</span> <span class="fu">summary</span>(corPredMeans)</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a> <span class="co">#   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a> <span class="co"># 0.9493  0.9798  0.9928  0.9855  0.9960  0.9981 </span></span></code></pre></div>
<p>The lowest corPredMeans was 0.95 for the SELIND, perhaps unsurprisingly.</p>
<p>Mean corPredMeans=0.9855. I think we are good to go.</p>
<p>Make a plot (below) to examine the scaling of predMean_medium vs. predMean_full to be sure that combining predMean_full with predSD_medium is safe. Seems like it. Everything is on the same scale as expected.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># crossPreds_full$tidyPreds[[1]] %&gt;% </span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   rename(predMean_full=predMean) %&gt;% </span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   left_join(crossPreds_medium$tidyPreds[[1]] %&gt;% </span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co">#               rename(predMean_medium=predMean)) %&gt;% </span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   ggplot(aes(x=predMean_full,y=predMean_medium)) + </span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_point() + </span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_abline(slope=1,color=&#39;darkred&#39;) +</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   facet_wrap(~Trait, scales=&#39;free&#39;)</span></span></code></pre></div>
<p>Recompute <strong>predUsefulness</strong> using <strong>predMean_full</strong> before saving to disk.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>crossPreds_full<span class="sc">$</span>tidyPreds[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(crossPreds_medium<span class="sc">$</span>tidyPreds[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rename</span>(<span class="at">predMean_medium=</span>predMean)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">predUsefulness=</span>predMean<span class="sc">+</span>(<span class="fl">2.0627128</span><span class="sc">*</span>predSD),</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">sireGroup=</span><span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS13&quot;</span>,</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS14&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS14&quot;</span>,</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS15&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS15&quot;</span>,</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS16&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS16&quot;</span>,</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS17&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS17&quot;</span>,</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS18&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS18&quot;</span>,</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS19&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS19&quot;</span>,</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS20&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS20&quot;</span>,</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>                                    <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13|TMS14|TMS15|TMS16|TMS17|TMS18|TMS19|TMS20&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;PreGS&quot;</span>),</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">damGroup=</span><span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS13&quot;</span>,</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS14&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS14&quot;</span>,</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS15&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS15&quot;</span>,</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS16&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS16&quot;</span>,</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS17&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS17&quot;</span>,</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS18&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS18&quot;</span>,</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS19&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS19&quot;</span>,</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS20&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS20&quot;</span>,</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>                            <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13|TMS14|TMS15|TMS16|TMS17|TMS18|TMS19|TMS20&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;PreGS&quot;</span>),</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">CrossGroup=</span><span class="fu">paste0</span>(sireGroup,<span class="st">&quot;x&quot;</span>,damGroup)) <span class="sc">%&gt;%</span></span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(<span class="fu">contains</span>(<span class="st">&quot;Group&quot;</span>),<span class="at">.before =</span> <span class="st">&quot;Nsegsnps&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(predMean,<span class="at">.before =</span> <span class="st">&quot;predMean_medium&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(predOf,Trait,<span class="fu">desc</span>(predUsefulness)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.csv</span>(.,<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicMatePredictions_2021Aug10.csv&quot;</span>),</span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> F)</span></code></pre></div>
</div>
<div id="september-2021" class="section level2">
<h2>September 2021</h2>
<p><strong>NOTE:</strong> For cross predictions, check that the predMean from full and medium set are highly correlated. As long as that is true, combine the predMean from full set with pred var from medium set.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>crossPreds_full<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicMatePredictions_full_set_2021Sep13.rds&quot;</span>))</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>crossPreds_medium<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicMatePredictions_medium_set_chunk1_2021Sep13.rds&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(<span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicMatePredictions_medium_set_chunk2_2021Sep13.rds&quot;</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(<span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicMatePredictions_medium_set_chunk3_2021Sep13.rds&quot;</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(tidyPreds) <span class="sc">%&gt;%</span> </span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(tidyPreds)</span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>crossPreds_full<span class="sc">$</span>tidyPreds[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">predMean_full=</span>predMean) <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(crossPreds_medium <span class="sc">%&gt;%</span> </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rename</span>(<span class="at">predMean_medium=</span>predMean)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(predOf,Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corPredMeans=</span><span class="fu">cor</span>(predMean_full,predMean_medium),<span class="at">.groups =</span> <span class="st">&#39;drop&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(corPredMeans)) <span class="sc">%$%</span> <span class="fu">summary</span>(corPredMeans)</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Joining, by = c(&quot;sireID&quot;, &quot;damID&quot;, &quot;predOf&quot;, &quot;Trait&quot;)</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="co">#    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="co">#  0.9585  0.9732  0.9863  0.9818  0.9889  0.9964 </span></span></code></pre></div>
<p>The lowest corPredMeans was 0.95 for the SELIND, perhaps unsurprisingly.</p>
<p>Mean corPredMeans=0.98. I think we are good to go.</p>
<p>Make a plot (below) to examine the scaling of predMean_medium vs. predMean_full to be sure that combining predMean_full with predSD_medium is safe. Seems like it. Everything is on the same scale as expected.</p>
<p>Recompute <strong>predUsefulness</strong> using <strong>predMean_full</strong> before saving to disk.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>crossPreds_full<span class="sc">$</span>tidyPreds[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(crossPreds_medium <span class="sc">%&gt;%</span> </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rename</span>(<span class="at">predMean_medium=</span>predMean)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">predUsefulness=</span>predMean<span class="sc">+</span>(<span class="fl">2.0627128</span><span class="sc">*</span>predSD),</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">sireGroup=</span><span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS13&quot;</span>,</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS14&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS14&quot;</span>,</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS15&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS15&quot;</span>,</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS16&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS16&quot;</span>,</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS17&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS17&quot;</span>,</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS18&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS18&quot;</span>,</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS19&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS19&quot;</span>,</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS20&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;TMS20&quot;</span>,</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>                                    <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13|TMS14|TMS15|TMS16|TMS17|TMS18|TMS19|TMS20&quot;</span>,sireID)<span class="sc">~</span><span class="st">&quot;PreGS&quot;</span>),</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">damGroup=</span><span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS13&quot;</span>,</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS14&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS14&quot;</span>,</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS15&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS15&quot;</span>,</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS16&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS16&quot;</span>,</span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS17&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS17&quot;</span>,</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS18&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS18&quot;</span>,</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS19&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS19&quot;</span>,</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">grepl</span>(<span class="st">&quot;TMS20&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;TMS20&quot;</span>,</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>                            <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;2013_|TMS13|TMS14|TMS15|TMS16|TMS17|TMS18|TMS19|TMS20&quot;</span>,damID)<span class="sc">~</span><span class="st">&quot;PreGS&quot;</span>),</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">CrossGroup=</span><span class="fu">paste0</span>(sireGroup,<span class="st">&quot;x&quot;</span>,damGroup)) <span class="sc">%&gt;%</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(<span class="fu">contains</span>(<span class="st">&quot;Group&quot;</span>),<span class="at">.before =</span> <span class="st">&quot;Nsegsnps&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(predMean,<span class="at">.before =</span> <span class="st">&quot;predMean_medium&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(predOf,Trait,<span class="fu">desc</span>(predUsefulness)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.csv</span>(.,<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;genomicMatePredictions_2021Sep18.csv&quot;</span>),</span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> F)</span></code></pre></div>
</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<p>See <a href="07-Results.html">Results</a>: Home for plots and summary tables.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
