<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>IITA 2021 Genomic Prediction and Mate Selection</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">IITA_2021GS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Imputation
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="convertDCas21_6038_ToVCF.html">Conversion DArT report DCas21_6038 to VCF</a>
    </li>
    <li>
      <a href="ImputeDCas21_6038.html">Impute DCas21_6038</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Preliminary Steps
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-cleanTPdata.html">Data Cleaning</a>
    </li>
    <li>
      <a href="02-GetBLUPs.html">Get BLUPs</a>
    </li>
    <li>
      <a href="03-validatePedigree.html">Verify pedigree records</a>
    </li>
    <li>
      <a href="04-PreprocessDataFiles.html">Preprocessing input data</a>
    </li>
  </ul>
</li>
<li>
  <a href="05-CrossValidation.html">Cross-validation</a>
</li>
<li>
  <a href="06-GenomicPredictions.html">Genomic (mate) predictions</a>
</li>
<li>
  <a href="07-Results.html">Results</a>
</li>
<li>
  <a href="inputsForSimulation.html">Inputs for Sims</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Other
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="about.html">About</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wolfemd/IITA_2021GS">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">IITA 2021 Genomic Prediction and Mate Selection</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-19
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>IITA_2021GS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomwolfemdIITA2021GStreeb0425f3195384021a62a6e8b20893fe512fdb025targetblankb0425f3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/wolfemd/IITA_2021GS/tree/b0425f3195384021a62a6e8b20893fe512fdb025" target="_blank">b0425f3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomwolfemdIITA2021GStreeb0425f3195384021a62a6e8b20893fe512fdb025targetblankb0425f3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/wolfemd/IITA_2021GS/tree/b0425f3195384021a62a6e8b20893fe512fdb025" target="_blank">b0425f3</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  data/DatabaseDownload_2021Aug08/
    Untracked:  data/DatabaseDownload_2021May04/
    Untracked:  data/GBSdataMasterList_31818.csv
    Untracked:  data/IITA_GBStoPhenoMaster_33018.csv
    Untracked:  data/NRCRI_GBStoPhenoMaster_40318.csv
    Untracked:  data/PedigreeGeneticGainCycleTime_aafolabi_01122020.xls
    Untracked:  data/Report-DCas21-6038/
    Untracked:  data/blups_forGP.rds
    Untracked:  data/chr1_RefPanelAndGSprogeny_ReadyForGP_72719.fam
    Untracked:  data/dosages_IITA_2021Aug09.rds
    Untracked:  data/haps_IITA_2021Aug09.rds
    Untracked:  data/recombFreqMat_1minus2c_2021Aug02.qs
    Untracked:  output/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/b0425f3195384021a62a6e8b20893fe512fdb025/analysis/index.Rmd" target="_blank">b0425f3</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-19
</td>
<td>
Integrate and publish results of an analysis to decide on inputs for
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/e029efce61084b2a8270f54742e9daf9330d9366/docs/index.html" target="_blank">e029efc</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/efebeaba8ea90c30c66b26d4187dcf979711f05a/analysis/index.Rmd" target="_blank">efebeab</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-12
</td>
<td>
Cross-validation and genomic mate predictions complete. All results updated.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/1c03315dbe60717a6f021e182f0b1bf5d054bc65/docs/index.html" target="_blank">1c03315</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/e4df79f49f8c6e7bd1d5c33df830bbfdd4419df9/analysis/index.Rmd" target="_blank">e4df79f</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-11
</td>
<td>
Completed IITA_2021GS pipeline including imputation and genomic prediction. Last bit of cross-validation and cross-prediction finishes in 24 hrs.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/a3150ab9cdb2190a6a26055c762d3d058eda5b3c/docs/index.html" target="_blank">a3150ab</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/6f2057f628fafc6049a09fd9ff3df2605653ae47/analysis/index.Rmd" target="_blank">6f2057f</a>
</td>
<td>
wolfemd
</td>
<td>
2021-08-09
</td>
<td>
Publish project. Imputation completed. Run and complete ‘cleanTPdata’ step.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/934141c48eff754aa674925be5357d0a2b8be7d3/docs/index.html" target="_blank">934141c</a>
</td>
<td>
wolfemd
</td>
<td>
2021-07-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/cc1eb4b5bb7f38ca04846562d27a3bd4951c6c09/docs/index.html" target="_blank">cc1eb4b</a>
</td>
<td>
wolfemd
</td>
<td>
2021-07-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/772750a8e35e6472113d69e6bc1eff0101aa137f/analysis/index.Rmd" target="_blank">772750a</a>
</td>
<td>
wolfemd
</td>
<td>
2021-07-14
</td>
<td>
DirDom model and selection index calc fully integrated functions.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/5e45aac791be42c4e432387023b49a41d58ecfe6/docs/index.html" target="_blank">5e45aac</a>
</td>
<td>
wolfemd
</td>
<td>
2021-06-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/df7a366d1b6d6c3579d2f4fe32935e1cbb7f4de4/docs/index.html" target="_blank">df7a366</a>
</td>
<td>
wolfemd
</td>
<td>
2021-06-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/c28400faf855cff1720a3566c8acf23641ee2ebf/analysis/index.Rmd" target="_blank">c28400f</a>
</td>
<td>
wolfemd
</td>
<td>
2021-06-10
</td>
<td>
github link added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2021GS/e66bdadd9561b8bf0e6bd6d77bd02621f05a9341/docs/index.html" target="_blank">e66bdad</a>
</td>
<td>
wolfemd
</td>
<td>
2021-06-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/a8452bafa0ab7a515967a022ca5c4b2638bb97ce/analysis/index.Rmd" target="_blank">a8452ba</a>
</td>
<td>
wolfemd
</td>
<td>
2021-06-10
</td>
<td>
Initial build of the entire page upon completion of all
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2021GS/blob/8a0c50eca5be95c975c345dc7f41b41863123404/analysis/index.Rmd" target="_blank">8a0c50e</a>
</td>
<td>
wolfemd
</td>
<td>
2021-05-04
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<ul>
<li>I recently (Aug 3, 2021) completed thorough testing to develop a protocol for genomic <em>mate</em> selection in NextGen Cassava and will implement it here for the first time in practice! Check <a href="https://wolfemd.github.io/implementGMSinCassava/">here</a> to see the implementation tests / code-base development documented.</li>
<li>Subsequently, I have developed my entire code base into an R package <a href="https://wolfemd.github.io/genomicMateSelectR/index.html">genomicMateSelectR</a>. The package is fully documented (in rough draft version), but doesn’t yet include a tutorial / vignette.</li>
<li>The IITA DArTseqLD report (DCas21_6038), which contains the GS C5 (i.e. progeny of crosses made in 2020, “TMS20F”), was recently (July 19, 2021) released.</li>
<li>The imputations of DCas21_6038, genomic predictions and mate selection that follows will leverage <a href="https://wolfemd.github.io/genomicMateSelectR/index.html">genomicMateSelectR</a> functions and will form the principal example of their use. Install it e.g. <code>devtools::install_github("wolfemd/genomicMateSelectR", ref = 'master')</code> .</li>
</ul>
<p><a href="07-Results.html"><strong>See the Results here!</strong></a></p>
<div id="august-2021" class="section level1">
<h1>August 2021</h1>
<div id="imputation-of-dcas21_6038" class="section level2">
<h2>Imputation of DCas21_6038</h2>
<p><strong>Steps</strong>:</p>
<ol style="list-style-type: decimal">
<li><a href="convertDCas21_6038_ToVCF.html">Convert DCas21-6038 report to VCF for imputation</a>:</li>
<li><a href="ImputeDCas21_6038.html">Impute DCas21-6038</a>: with West Africa reference panel merged with additional GS progeny (IITA TMS18)</li>
</ol>
<p><strong>Files</strong>: Access on Cassavabase FTP server here, use “Guest” credentials</p>
<ul>
<li><strong>RefPanel VCF filename:</strong> <code>chr*_RefPanelAndGSprogeny_ReadyForGP_72719.vcf.gz</code><br />
(<a href="ftp://ftp.cassavabase.org/marnin_datasets/nextgenImputation2019/ImputationStageIII_72619/"><strong>Click here for FTP server location</strong></a>)</li>
<li><strong>Imputed filename:</strong> <code>output/chr*_DCas21_6038_WA_REFimputed.vcf.gz</code><br />
(<a href="ftp://ftp.cassavabase.org/marnin_datasets/IITA_2021GS/"><strong>Click here for FTP server location</strong></a>)</li>
<li><strong>Post-impute filtered filename:</strong> <code>output/chr*_DCas21_6038_WA_REFimputedAndFiltered.vcf.gz</code><br />
(<a href="ftp://ftp.cassavabase.org/marnin_datasets/IITA_2021GS/"><strong>Click here for FTP server location</strong></a>)</li>
<li><strong>All chrom. combined, RefPanel and DCas21_6038 VCF merged</strong> <code>output/AllChrom_RefPanelAndGSprogeny_ReadyForGP_2021Aug08.vcf.gz</code><br />
(<a href="ftp://ftp.cassavabase.org/marnin_datasets/IITA_2021GS/"><strong>Click here for FTP server location</strong></a>)</li>
</ul>
</div>
<div id="preliminary-data-steps" class="section level2">
<h2>Preliminary data steps</h2>
<ol style="list-style-type: decimal">
<li><p><a href="01-cleanTPdata.html">Prepare training dataset</a>: Download data from DB, “Clean” and format DB data.</p></li>
<li><p><a href="02-GetBLUPs.html">Get BLUPs combining all trial data</a>: Combine data from all trait-trials to get BLUPs for downstream genomic prediction. Fit mixed-model to multi-trial dataset and extract BLUPs, de-regressed BLUPs and weights. Include two rounds of outlier removal.</p></li>
<li><p><a href="03-validatePedigree.html">Validate the pedigree obtained from cassavabase</a>: Before setting up a cross-validation scheme for predictions that depend on a correct pedigree, add a basic verification step to the pipeline. Not trying to fill unknown relationships or otherwise correct the pedigree. Assess evidence that relationship is correct, remove if incorrect.</p></li>
<li><p><a href="04-PreprocessDataFiles.html">Preprocess data files</a>: Prepare haplotype and dosage matrices, GRMs, pedigree and BLUPs, genetic map <em>and</em> recombination frequency matrix, for use in predictions.</p></li>
</ol>
</div>
<div id="genomic-mate-predictions" class="section level2">
<h2>Genomic (mate) predictions</h2>
<ol start="5" style="list-style-type: decimal">
<li><p><a href="05-CrossValidation.html">Parent-wise and standard cross-validation</a>: estimate selection index (and component trait) prediction accuracies using the direction-dominance (<strong>DirDom</strong>) model.</p>
<ul>
<li><p>Additionally, <a href="https://wolfemd.github.io/IITA_2021GS/05-CrossValidation.html#Optimize_marker_density_for_speed">check accuracy and similarity of predictions at reduced marker density</a>: Cross-variance prediction is slow, but significant speed gains can be made by using fewer markers. Faster predictions will mean more crosses can be predicted and considered.</p>
<ul>
<li><strong><em>If,</em></strong> accuracy and <span class="math inline">\(cor_{preds}(All\_SNPs, Reduced\_Set)\)</span> are similar based on both kinds of cross-validation, proceed to make cross-variance predictions with reduced marker set; possibly use full marker set for cross-mean predictions.</li>
</ul></li>
<li><p><a href="https://wolfemd.github.io/IITA_2021GS/07-Results.html#Prediction_accuracy_estimates">Click here to see the results</a><strong>!</strong></p></li>
</ul></li>
<li><p><a href="06-GenomicPredictions.html">Genomic predictions</a>: First, predict of individual GEBV/GETGV for all selection candidates using all available data and return marker effects for use downstream. Next, Select a top set of candidate parents, for whom we would like to predict cross performances. Finally, predict all pairwise crosses of candidate parents and evaluate them for genomic mate selection. Select the top crosses and plant a crossing nursery with the parents indicated.</p>
<ul>
<li><a href="https://wolfemd.github.io/IITA_2021GS/07-Results.html#Genomic_Predictions">Click here to see the results</a><strong>!</strong></li>
</ul></li>
<li><p><a href="07-Results.html">Results and recommendations</a>: Home for all plots, summary tables, conclusions and recommendations.</p></li>
</ol>
</div>
<div id="empricial-inputs-for-simulations" class="section level2">
<h2>Empricial inputs for simulations</h2>
<p>See <a href="inputsForSimulation.html">here</a> for an analysis of IITA trial data to derive empirical inputs for <a href="https://wolfemd.github.io/BreedingSchemeOpt/">breeding scheme simulations</a>. I analyze trial-by-trial, picking up after the <a href="01-cleanTPdata.html">data cleaning step</a> above, to measure selection error and associate it with plot-size.</p>
</div>
</div>
<div id="data-and-code-repository-access" class="section level1">
<h1>Data and code repository access</h1>
<p><a href="ftp://ftp.cassavabase.org/marnin_datasets/IITA_2021GS/">CLICK HERE FOR ACCESS TO THE FULL REPOSITORY</a><br />
(select “Guest” credentials when prompted by the Cassavabase FTP server)</p>
<p>or</p>
<p><a href="https://github.com/wolfemd/IITA_2021GS/"><strong>DOWNLOAD FROM GitHub*</strong></a></p>
<p>*GitHub only hosts files max 50 Mb.</p>
</div>
<div id="key-directories-and-file-names" class="section level1">
<h1>Key directories and file names</h1>
<ol style="list-style-type: decimal">
<li><code>data/</code>: raw data (e.g. unimputed SNP data)</li>
<li><code>output/</code>: outputs (e.g. imputed SNP data)</li>
<li><code>analysis/</code>: most code and workflow documented in <strong>.Rmd</strong> files</li>
<li><code>docs/</code>: compiled <strong>.html</strong>, “knitted” from <strong>.Rmd</strong></li>
<li><code>code/</code>: supporting functions sourced in <code>analysis/*.Rmd</code>’s.</li>
</ol>
<p><strong>FILES OF INTEREST</strong>: everything is in the <code>output/</code> sub-directory (<a href="ftp://ftp.cassavabase.org/marnin_datasets/IITA_2021GS/output/">click here</a> and select “Guest” credentials when prompted by the Cassavabase FTP server).</p>
<ul>
<li><p><strong>GEBVs for parent selection and GETGVs for variety advancement:</strong></p>
<ul>
<li><a href="ftp://ftp.cassavabase.org/marnin_datasets/IITA_2021GS/output/genomicPredictions_full_set_2021Aug09.csv">download DirDom model genomic predictions</a></li>
</ul></li>
<li><p><strong>Predicted means, variances and usefulness of crosses among top parents:</strong></p>
<ul>
<li><a href="ftp://ftp.cassavabase.org/marnin_datasets/IITA_2021GS/output/genomicMatePredictions_2021Aug10.csv">download DirDom model genomic mate predictions</a></li>
</ul></li>
<li><p>Kinship matrices, dosages, haplotype matrix, recombination frequency matrix, genetic map files</p></li>
</ul>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
