---
title: "Genomic predictions"
author: "Marnin Wolfe"
date: "2021-July-08"
output: 
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = F, 
                      eval = FALSE, # <- NOTE THAT EVAL SET TO FALSE!
                      tidy='styler', tidy.opts=list(strict=FALSE,width.cutoff=100), highlight=TRUE)
```

# Previous step

5.  [Parent-wise cross-validation](05-CrossValidation.html): Compute parent-wise cross-validation folds using the validated pedigree. Fit models to get marker effects and make subsequent predictions of cross means and (co)variances.

# Current steps

1.  **Genomic prediction of clone GEBV/GETGV.** Fit GBLUP model, using genotypic add-dom partition. NEW: modelType="DirDom", include genome-wide inbreeding effect in GEBV/GETGV predictions *after* backsolving SNP effects. For all models, extract GBLUPs and backsolve SNP effects for use in cross usefulness predictions (mean+variance predictions). **ALSO NEW**: selection index predictions.

2.  **Genomic prediction of cross** $UC_{parent}$ and $UC_{variety}$. Rank potential parents on SI. Predict all possible crosses of some portion of best parents.

# Genomic prediction of clone GEBV/GETGV

Operate R within a singularity Linux shell within a screen shell.

```{bash set-up R environment, eval=F}
# 1) start a screen shell 
screen; # or screen -r if re-attaching...
# 2) start the singularity Linux shell inside that
singularity shell /workdir/$USER/rocker.sif; 
# Project directory, so R will use as working dir.
cd /home/mw489/implementGMSinCassava/;
# 3) Start R
R
```

Load input data
```{r load input data for GP, eval=F}
require(tidyverse); require(magrittr); 
# 5 threads per Rsession for matrix math (openblas)
RhpcBLASctl::blas_set_num_threads(5)
# GENOMIC MATE SELECTION FUNCTIONS
source(here::here("code","gmsFunctions.R")) 
source(here::here("code","predCrossVar.R")) 

# GENOMIC RELATIONSHIP MATRICES (GRMS)
grms<-list(A=readRDS(file=here::here("output","kinship_A_IITA_2021May13.rds")),
           D=readRDS(file=here::here("output","kinship_domGenotypic_IITA_2021July5.rds")))

# DOSAGE MATRIX
dosages<-readRDS(file=here::here("data",
                                 "dosages_IITA_filtered_2021May13.rds"))

# BLUPs
blups<-readRDS(file=here::here("data","blups_forCrossVal.rds")) %>% 
  dplyr::select(-varcomp) %>% 
  rename(TrainingData=blups) # for compatibility with downstream functions

# SELECTION INDEX WEIGHTS
## from IYR+IK
## note that not ALL predicted traits are on index
SIwts<-c(logFYLD=20,
         HI=10,
         DM=15,
         MCMDS=-10,
         logRTNO=12,
         logDYLD=20,
         logTOPYLD=15,
         PLTHT=10) 
```

Run the "AD" and "DirDom" modelTypes built into `runGenomicPredictions()`. 

Output contains both GBLUPs for selection of indivuals _and_ SNP effects to use as input for prediction of cross usefulness and subsequent mate selection.

```{r run genomic predictions}
start<-proc.time()[3]
gpreds_ad<-runGenomicPredictions(modelType="AD",selInd=TRUE, SIwts=SIwts,
                                 getMarkEffs=TRUE,
                                 returnPEV=FALSE,
                                 blups=blups,grms=grms,dosages=dosages,
                                 ncores=20,nBLASthreads=5)
runtime<-proc.time()[3]-start; runtime/60
saveRDS(gpreds_ad,file = here::here("output","genomicPredictions_ModelAD.rds"))

gpreds_dirdom<-runGenomicPredictions(modelType="DirDom",selInd=TRUE, SIwts=SIwts,
                                     getMarkEffs=TRUE,
                                     returnPEV=FALSE,
                                     blups=blups,grms=grms,dosages=dosages,
                                     ncores=20,nBLASthreads=5)
runtime<-proc.time()[3]-start; runtime/60
saveRDS(gpreds_dirdom,file = here::here("output","genomicPredictions_ModelDirDom.rds"))

```

# Genomic mate predictions: cross usefulness

1. Use GBLUPs on SELIND to choose some to fraction of the clones.
2. For those selected parents, predict the SELIND usefulness for all pairwise matings.

```{bash, eval=F}
# 1) start a screen shell 
screen; # or screen -r if re-attaching...
# 2) start the singularity Linux shell inside that
#singularity shell /workdir/$USER/rocker.sif; 
singularity shell ~/rocker2.sif; 
# Project directory, so R will use as working dir.
cd /home/mw489/implementGMSinCassava/;
# 3) Start R
R
```

### Load input data
```{r load inputs for predictCrosses, eval=F}
require(tidyverse); require(magrittr); 
# 5 threads per Rsession for matrix math (openblas)
#RhpcBLASctl::blas_set_num_threads(5)
# GENOMIC MATE SELECTION FUNCTIONS
source(here::here("code","gmsFunctions.R")) 
source(here::here("code","predCrossVar.R")) 

# DOSAGE MATRIX
dosages<-readRDS(file=here::here("data",
                                 "dosages_IITA_filtered_2021May13.rds"))

# RECOMBINATION FREQUENCY MATRIX
recombFreqMat<-readRDS(file=here::here("data",
                                       "recombFreqMat_1minus2c_2021May13.rds"))
# HAPLOTYPE MATRIX
## keep only haplos for parents-in-the-pedigree
## those which will be used in prediction, saves memory
haploMat<-readRDS(file=here::here("data","haps_IITA_filtered_2021May13.rds"))
parents<-union(ped$sireID,ped$damID) 
parenthaps<-sort(c(paste0(parents,"_HapA"),
                   paste0(parents,"_HapB")))
haploMat<-haploMat[parenthaps,colnames(recombFreqMat)]

# SELECTION INDEX WEIGHTS
## from IYR+IK
## note that not ALL predicted traits are on index
SIwts<-c(logFYLD=20,
         HI=10,
         DM=15,
         MCMDS=-10,
         logRTNO=12,
         logDYLD=20,
         logTOPYLD=15,
         PLTHT=10) 

# SNP EFFECTS - FROM PREVIOUS STEP 
### Two models AD and DirDom
gpreds_ad<-readRDS(file = here::here("output","genomicPredictions_ModelAD.rds"))
gpreds_dirdom<-readRDS(file = here::here("output","genomicPredictions_ModelDirDom.rds"))
### Cor between SELIND GEBV and GETGV between models?
# left_join(gpreds_ad$gblups[[1]] %>% select(GID,predOf,SELIND) %>% rename(SELINDad=SELIND),
#           gpreds_dirdom$gblups[[1]] %>% select(GID,predOf,SELIND) %>% rename(SELINDdirdom=SELIND)) %>% 
#   group_by(predOf) %>% summarize(SELIND_corModels=cor(SELINDad,SELINDdirdom))
# predOf SELIND_corModels
# GEBV	0.9890091			
# GETGV	0.9373170	

# SELECTION OF CROSSES-TO-BE-PREDICTED
nParentsToSelect<-100
union_bestGEBVandGETGVdirdom<-union(gpreds_dirdom$gblups[[1]] %>% 
                                         filter(predOf=="GEBV") %>% 
                                         arrange(desc(SELIND)) %>% 
                                         slice(1:nParentsToSelect) %$% GID,
                                       gpreds_dirdom$gblups[[1]] %>% 
                                         filter(predOf=="GETGV") %>% 
                                         arrange(desc(SELIND)) %>% 
                                         slice(1:nParentsToSelect) %$% GID)
union_bestGEBVandGETGVad<-union(gpreds_ad$gblups[[1]] %>% 
                                         filter(predOf=="GEBV") %>% 
                                         arrange(desc(SELIND)) %>% 
                                         slice(1:nParentsToSelect) %$% GID,
                                       gpreds_ad$gblups[[1]] %>% 
                                         filter(predOf=="GETGV") %>% 
                                         arrange(desc(SELIND)) %>% 
                                         slice(1:nParentsToSelect) %$% GID)
union_bestGEBVandGETGV<-union(union_bestGEBVandGETGVdirdom,
                              union_bestGEBVandGETGVad)
length(union_bestGEBVandGETGV) 
# [1] 121 parents in top nParentsToSelect on SELIND for GEBV/GETGV - DirDom/AD model.

CrossesToPredict<-crosses2predict(union_bestGEBVandGETGV)
nrow(CrossesToPredict)
# [1] 7381 
```

Run the "AD" and "DirDom" modelTypes.  

### Testing
```{r debug inputs}
# modelType="DirDom";
# stdSelInt = 2.67; 
# selInd=TRUE;
# set.seed(111);
# SIwts=SIwts[sample(1:length(SIwts),3,replace = F)];
# 
# # predict first 3 crosses only for speed
# CrossesToPredict=CrossesToPredict %>% slice(1:3);
# 
# # sample 1000 SNPs to make preds go faster
# set.seed(22222)
# snps2keep<-sample(colnames(dosages),size = 1000,replace = F)
# 
# dosages=dosages[,snps2keep];
# haploMat=haploMat[,snps2keep];
# recombFreqMat=recombFreqMat[snps2keep,snps2keep];
# ncores=15;nBLASthreads=5
# 
# # model DirDom
# snpeffs=gpreds_dirdom$genomicPredOut[[1]];
# snpeffs %<>% 
#   filter(Trait %in% names(SIwts)) %>% 
#   mutate(across(contains("snpeff"),~map(.,~.[snps2keep,,drop=F])))
```
```{r debugged version of predictCrosses}
# predictCrosses<-function(modelType,
#                          stdSelInt = 2.67,
#                          selInd,SIwts = NULL,
#                          CrossesToPredict,
#                          snpeffs,dosages,
#                          haploMat,recombFreqMat,
#                          ncores=1,nBLASthreads=NULL){
#   ## Format SNP effect matrices ~~~~~~~~~~~~~~~~~~~~~~~~
# 
#   AlleleSubEffectList<-snpeffs$allelesubsnpeff %>%
#     `names<-`(.,snpeffs$Trait) %>%
#     map(.,~t(.))
# 
#   if(modelType=="AD"){
#     # DomDevEffects for model "AD" to predict VarTGV = VarBV + VarDD
#     DomDevEffectList=snpeffs$domdevsnpeff %>%
#       `names<-`(.,snpeffs$Trait) %>%
#       map(.,~t(.)) }
# 
#   if(modelType=="DirDom"){
#     # AddEffectList + DomEffectList --> VarTGV; AlleleSubEffectList --> VarBV;
#     AddEffectList<-snpeffs$addsnpeff %>%
#       `names<-`(.,snpeffs$Trait) %>%
#       map(.,~t(.))
#     DomEffectList<-snpeffs$domsnpeff %>%
#       `names<-`(.,snpeffs$Trait) %>%
#       map(.,~t(.)) }
# 
#   ## Predict cross variances ~~~~~~~~~~~~~~~~~~~~~~~~
#   print("Predicting cross variance parameters")
#   if(modelType=="A"){
#     predictedvars<-predCrossVars(CrossesToPredict=CrossesToPredict,
#                                  AddEffectList=AlleleSubEffectList,
#                                  modelType="A",
#                                  haploMat=haploMat,
#                                  recombFreqMat=recombFreqMat,
#                                  ncores=ncores,nBLASthreads=nBLASthreads) %>%
#       unnest(predVars) %>%
#       mutate(predOf="VarBV") }
#   if(modelType=="AD"){
#     predictedvars<-predCrossVars(CrossesToPredict=CrossesToPredict,
#                                  AddEffectList=AlleleSubEffectList,
#                                  DomEffectList=DomDevEffectList,
#                                  modelType="AD",
#                                  haploMat=haploMat,
#                                  recombFreqMat=recombFreqMat,
#                                  ncores=ncores,nBLASthreads=nBLASthreads)
#     predictedvars %<>%
#       unnest(predVars) %>%
#       mutate(predOf=ifelse(predOf=="VarA","VarBV","VarDD"))
#   }
#   if(modelType=="DirDom"){
#     predictedvarTGV<-predCrossVars(CrossesToPredict=CrossesToPredict,
#                                    AddEffectList=AddEffectList,
#                                    DomEffectList=DomEffectList,
#                                    modelType="AD", # no "DirDom" model in predCrossVars() nor is it needed
#                                    haploMat=haploMat,
#                                    recombFreqMat=recombFreqMat,
#                                    ncores=ncores,nBLASthreads=nBLASthreads)
#     predictedvarBV<-predCrossVars(CrossesToPredict=CrossesToPredict,
#                              AddEffectList=AlleleSubEffectList,
#                              DomEffectList=NULL,
#                              modelType="A", # no "DirDom" model in predCrossVars() nor is it needed
#                              haploMat=haploMat,
#                              recombFreqMat=recombFreqMat,
#                              ncores=ncores,nBLASthreads=nBLASthreads)
#     predictedvars<-predictedvarBV %>%
#       unnest(predVars) %>%
#       mutate(predOf="VarBV") %>%
#       bind_rows(predictedvarTGV %>%
#                   unnest(predVars)) }
# 
#   ## Predict cross means ~~~~~~~~~~~~~~~~~~~~~~~~
#   print("Predicting cross means")
#   ### predict MeanBVs
#   predictedmeans<-predCrossMeans(AddEffectList=AlleleSubEffectList,
#                                  CrossesToPredict=CrossesToPredict,
#                                  doseMat=dosages,
#                                  ncores=ncores,
#                                  nBLASthreads=nBLASthreads,
#                                  predType="BV")
#   if(modelType=="AD"){
#     ### DO NOT predict MeanTGV ~but~ duplicate MeanBV as MeanTGV prediction
#     ### there IS predVarTGV for this model, output predUC-TGV (i.e. UC_variety)
#     predictedmeans %<>%
#       bind_rows(predictedmeans %>% mutate(predOf="TGV")) }
# 
#   if(modelType=="DirDom"){
#     ### predict MeanTGVs
#     ####  Prediction of MeanTGV is only available for the DirDom model
#     #### or a model with "genotypic" additive-dominance SNP effects
#     #### As implemented, modelType="AD" is the "classical" partition (BVs+ DomDevs)
#     predictedmeans %<>%
#       bind_rows(predCrossMeans(AddEffectList=AddEffectList,
#                                DomEffectList=DomEffectList,
#                                CrossesToPredict=CrossesToPredict,
#                                doseMat=dosages,
#                                ncores=ncores,
#                                nBLASthreads=nBLASthreads,
#                                predType="TGV"))
#   }
# 
#   ## SIMPLIFIED, TIDY, CROSS-WISE OUTPUT ~~~~~~~~~~~~~~~~~~~~~~~~
#   # store raw mean and var preds
#   rawPreds<-list(predMeans=list(predictedmeans),
#                  predVars=list(predictedvars))
# 
#   ## tidy pred. means ~~~~~~
#   predictedmeans %<>%
#     mutate(predOf=gsub("Mean","",predOf),
#            Trait2=Trait) %>% # to match with variance pred. output
#     rename(Trait1=Trait) %>% # to match with variance pred. output
#     select(sireID,damID,predOf,Trait1,Trait2,predMean)
# 
#   ## tidy pred. vars ~~~~~~
#   predictedvars %<>%
#     select(sireID,damID,Nsegsnps,predOf,Trait1,Trait2,predVar) %>%
#     mutate(predOf=gsub("Var","",predOf))
#   if(modelType=="AD"){
#     predictedvars %<>%
#       filter(predOf=="BV") %>%
#       bind_rows(predictedvars %>%
#                   pivot_wider(names_from = "predOf",
#                               values_from = "predVar",
#                               names_prefix = "predVar") %>%
#                   mutate(predVar=predVarBV+predVarDD,
#                          predOf="TGV") %>%
#                   select(-predVarBV,-predVarDD))
#   }
#   if(modelType=="DirDom"){
#     predictedvars %<>%
#       filter(predOf=="BV") %>%
#       bind_rows(predictedvars %>%
#                   filter(predOf!="BV") %>%
#                   pivot_wider(names_from = "predOf",
#                               values_from = "predVar",
#                               names_prefix = "predVar") %>%
#                   mutate(predVar=predVarA+predVarD,
#                          predOf="TGV") %>%
#                   select(-predVarA,-predVarD))
#   }
# 
#   ## SELECTION INDEX MEANS AND VARIANCES ~~~~~~~~~~~~~~~~~~~~~~~~
#   #### Compute and add to tidy output, if requested
#   if(selInd){
#     print("Computing SELECTION INDEX means and variances.")
#     traits<-unique(predictedmeans$Trait1)
#     ## Compute Mean SELIND
#     predictedmeans %<>%
#       select(-Trait2) %>%
#       spread(Trait1,predMean) %>%
#       select(sireID,damID,predOf,all_of(traits)) %>%
#       mutate(SELIND=as.numeric((predictedmeans %>%
#                                   select(-Trait2) %>%
#                                   spread(Trait1,predMean) %>%
#                                   select(all_of(names(SIwts))) %>%
#                                   as.matrix(.))%*%SIwts)) %>%
#       relocate(SELIND, .after = predOf) %>%
#       pivot_longer(cols = c(SELIND,all_of(traits)),
#                    names_to = "Trait1",
#                    values_to = "predMean") %>%
#       mutate(Trait2=Trait1) %>%
#       select(sireID,damID,predOf,Trait1,Trait2,predMean)
#     ## Compute Var SELIND
#     require(furrr); plan(multisession, workers = ncores)
#     options(future.globals.maxSize=+Inf); options(future.rng.onMisuse="ignore")
# 
#     predictedvars %<>%
#       nest(predVars=c(Trait1,Trait2,predVar)) %>%
#       ## loop over each rep-fold-predOf-sireIDxdamID
#       mutate(predVars=future_map(predVars,function(predVars,...){
# 
#         gmat<-predVars %>%
#           pivot_wider(names_from = "Trait2",
#                       values_from = "predVar") %>%
#           column_to_rownames(var = "Trait1") %>%
#           as.matrix
#         gmat[lower.tri(gmat)]<-t(gmat)[lower.tri(gmat)]
#         gmat %<>% .[names(SIwts),names(SIwts)]
#         predSelIndVar<-SIwts%*%gmat%*%SIwts
#         ## add sel index predictions to component trait
#         ## var-covar predictions
# 
#         predVars<-tibble(Trait1="SELIND",Trait2="SELIND",
#                          predVar=as.numeric(predSelIndVar)) %>%
#           bind_rows(predVars)
#         return(predVars) })) %>%
#       unnest(predVars)
#     plan(sequential)
#   }
# 
#   ## USEFULNESS CRITERIA ~~~~~~~~~~~~~~~~~~~~~~~~
#   tidyPreds<-predictedvars %>%
#     inner_join(predictedmeans) %>%
#     rename(Trait=Trait1) %>%
#     select(sireID,damID,Nsegsnps,predOf,Trait,predMean,predVar) %>%
#     mutate(predSD=sqrt(predVar),
#            predUsefulness=predMean+(stdSelInt*predSD))
# 
#   predictions<-tibble(tidyPreds=list(tidyPreds),
#                       rawPreds=list(rawPreds))
#   return(predictions)
# }
``` 
```{r run tests of both modelTypes}
# set.seed(111);
# SIwts=SIwts[sample(1:length(SIwts),3,replace = F)];
# # predict first 3 crosses only for speed
# CrossesToPredict=CrossesToPredict %>% slice(1:3);
# # sample 1000 SNPs to make preds go faster
# set.seed(22222)
# snps2keep<-sample(colnames(dosages),size = 1000,replace = F)
# dosages=dosages[,snps2keep];
# haploMat=haploMat[,snps2keep];
# recombFreqMat=recombFreqMat[snps2keep,snps2keep];
# 
# # model DirDom
# snpeffs=gpreds_dirdom$genomicPredOut[[1]];
# snpeffs %<>%
#   filter(Trait %in% names(SIwts)) %>%
#   mutate(across(contains("snpeff"),~map(.,~.[snps2keep,,drop=F])))
# testpreds_dirdom<-predictCrosses(modelType="DirDom",
#                                  stdSelInt = 2.67, 
#                                  selInd=TRUE, SIwts=SIwts,
#                                  CrossesToPredict=CrossesToPredict,
#                                  snpeffs=snpeffs,
#                                  dosages=dosages,
#                                  haploMat=haploMat,recombFreqMat=recombFreqMat,
#                                  ncores=15,nBLASthreads=5)
# 
# 
# # model AD
# snpeffs=gpreds_ad$genomicPredOut[[1]];
# snpeffs %<>%
#   filter(Trait %in% names(SIwts)) %>%
#   mutate(across(contains("snpeff"),~map(.,~.[snps2keep,,drop=F])))
# testpreds_ad<-predictCrosses(modelType="AD",
#                              stdSelInt = 2.67, 
#                              selInd=TRUE, SIwts=SIwts,
#                              CrossesToPredict=CrossesToPredict,
#                              snpeffs=snpeffs,
#                              dosages=dosages,
#                              haploMat=haploMat,recombFreqMat=recombFreqMat,
#                              ncores=15,nBLASthreads=5)

```



### Full run 

cbsulm15 - July 25 - 7:30pm
```{r predict UC - top 121 parents - model DirDom}
start<-proc.time()[3]
crossPreds_dirdom<-predictCrosses(modelType="DirDom",stdSelInt = 2.67, 
                                  selInd=TRUE, SIwts=SIwts,
                                  CrossesToPredict=CrossesToPredict,
                                  snpeffs=gpreds_dirdom$genomicPredOut[[1]],
                                  dosages=dosages,
                                  haploMat=haploMat,recombFreqMat=recombFreqMat,
                                  ncores=20,nBLASthreads=5)
runtime<-proc.time()[3]-start; runtime/60
saveRDS(crossPreds_dirdom,file = here::here("output","genomicMatePredictions_top121parents_ModelDirDom.rds"))

```

cbsulm17 - July 25 at 7:30pm
```{r predict UC - top 121 parents - model AD}
start<-proc.time()[3]
crossPreds_ad<-predictCrosses(modelType="AD",stdSelInt = 2.67, 
                              selInd=TRUE, SIwts=SIwts,
                              CrossesToPredict=CrossesToPredict,
                              snpeffs=gpreds_ad$genomicPredOut[[1]],
                              dosages=dosages,
                              haploMat=haploMat,recombFreqMat=recombFreqMat,
                              ncores=20,nBLASthreads=5)
runtime<-proc.time()[3]-start; runtime/60
saveRDS(crossPreds_ad,file = here::here("output","genomicMatePredictions_top121parents_ModelAD.rds"))
```

# [TO DO] Write predictions to disk

Add genetic groups and tidy format

```{r}
# unique_gids<-predModelA %>%
#   dplyr::select(genomicPredOut) %>%
#   unnest(genomicPredOut) %>%
#   select(-varcomps) %>%
#   unnest(gblups) %$%
#   GID %>%
#   unique
# 
# c1a<-unique_gids %>%
#   grep("c1a",.,value = T,ignore.case = T) %>%
#   union(.,unique_gids %>%
#           grep("^F",.,value = T,ignore.case = T) %>%
#           grep("c1b",.,value = T,ignore.case = T,invert = T))
# c1b<-unique_gids%>% grep("c1b",.,value = T,ignore.case = T)
# c2a<-unique_gids %>%
#   grep("C2a",.,value = T,ignore.case = T) %>%
#   grep("NR17",.,value = T,ignore.case = T)
# c2b<-unique_gids %>%
#   grep("C2b",.,value = T,ignore.case = T) %>%
#   .[!. %in% c(c1a,c1b,c2a)]
# c3a<-unique_gids %>%
#   grep("C3a",.,value = T,ignore.case = T) %>%
#   .[!. %in% c(c1a,c1b,c2a,c2b)]
# c3b<-unique_gids %>%
#   grep("NR20C3",.,value = T,ignore.case = T) %>%
#   .[!. %in% c(c1a,c1b,c2a,c2b,c3a)]
# nrTP<-setdiff(unique_gids,unique(c(c1a,c1b,c2a,c2b,c3a,c3b)))
```

Write CSV's

```{r}
# ## Format and write GEBV
# predModelA %>% 
#   select(Trait,genomicPredOut) %>% 
#   unnest(genomicPredOut) %>% 
#   select(-varcomps) %>% 
#   unnest(gblups) %>% 
#   select(-GETGV,-contains("PEV")) %>%
#   spread(Trait,GEBV) %>% 
#   mutate(Group=case_when(GID %in% nrTP ~ "nrTP",
#                          GID %in% c1a ~ "C1a",
#                          GID %in% c1b ~ "C1b",
#                          GID %in% c2a ~ "C2a",
#                          GID %in% c2b ~ "C2b",
#                          GID %in% c3a ~ "C3a",
#                          GID %in% c3b ~ "C3b")) %>% 
#   select(Group,GID,any_of(traits)) %>% 
#   arrange(desc(Group)) %>% 
#   write.csv(., file = here::here("output","GEBV_NRCRI_ModelA_2021May03.csv"), row.names = F)
# 
# ## Format and write GETGV
# predModelADE %>% 
#   select(Trait,genomicPredOut) %>% 
#   unnest(genomicPredOut) %>% 
#   select(-varcomps) %>% 
#   unnest(gblups) %>% 
#   select(GID,Trait,GETGV) %>% 
#   spread(Trait,GETGV) %>% 
#   mutate(Group=case_when(GID %in% nrTP ~ "nrTP",
#                          GID %in% c1a ~ "C1a",
#                          GID %in% c1b ~ "C1b",
#                          GID %in% c2a ~ "C2a",
#                          GID %in% c2b ~ "C2b",
#                          GID %in% c3a ~ "C3a",
#                          GID %in% c3b ~ "C3b")) %>% 
#   select(Group,GID,any_of(traits)) %>% 
#   arrange(desc(Group)) %>% 
#   write.csv(., file = here::here("output","GETGV_NRCRI_ModelADE_2021May03.csv"), row.names = F)
# 
# ### Make a unified "tidy" long-form: 
# predModelA %>% 
#   select(Trait,genomicPredOut) %>% 
#   unnest(genomicPredOut) %>% 
#   select(-varcomps) %>% 
#   unnest(gblups) %>% 
#   select(-GETGV) %>% 
#   full_join(predModelADE %>% 
#               select(Trait,genomicPredOut) %>% 
#               unnest(genomicPredOut) %>% 
#               select(-varcomps) %>% 
#               unnest(gblups) %>% 
#               rename(GEBV_modelADE=GEBV,
#                      PEV_modelADE=PEVa) %>% 
#               select(-genomicPredOut)) %>% 
#   mutate(Group=case_when(GID %in% nrTP ~ "nrTP",
#                          GID %in% c1a ~ "C1a",
#                          GID %in% c1b ~ "C1b",
#                          GID %in% c2a ~ "C2a",
#                          GID %in% c2b ~ "C2b",
#                          GID %in% c3a ~ "C3a",
#                          GID %in% c3b ~ "C3b")) %>% 
#   relocate(Group,.before = GID) %>% 
#   write.csv(., file = here::here("output","genomicPredictions_NRCRI_2021May03.csv"), row.names = F)
```




# Results

See [Results](07-Results.html): Home for plots and summary tables.
