---
title: "Genomic predictions"
author: "Marnin Wolfe"
date: "2021-July-08"
output: 
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = F, 
                      tidy='styler', tidy.opts=list(strict=FALSE,width.cutoff=100), highlight=TRUE)
```

# Previous step

5.  [Parent-wise cross-validation](05-CrossValidation.html): Compute parent-wise cross-validation folds using the validated pedigree. Fit models to get marker effects and make subsequent predictions of cross means and (co)variances.

# Current steps

1.  **Genomic prediction of clone GEBV/GETGV.** Fit GBLUP model, using genotypic add-dom partition. NEW: modelType="DirDom", include genome-wide inbreeding effect in GEBV/GETGV predictions *after* backsolving SNP effects. For all models, extract GBLUPs and backsolve SNP effects for use in cross usefulness predictions (mean+variance predictions). **ALSO NEW**: selection index predictions.

2.  **Genomic prediction of cross** $UC_{parent}$ and $UC_{variety}$. Rank potential parents on SI. Predict all possible crosses of some portion of best parents.

# Genomic prediction of clone GEBV/GETGV

Operate R within a singularity Linux shell within a screen shell.

```{bash set-up R environment, eval=F}
# 1) start a screen shell 
screen; # or screen -r if re-attaching...
# 2) start the singularity Linux shell inside that
singularity shell /workdir/$USER/rocker.sif; 
# Project directory, so R will use as working dir.
cd /home/mw489/implementGMSinCassava/;
# 3) Start R
R
```

Load input data
```{r load input data for GP, eval=F}
require(tidyverse); require(magrittr); 
# 5 threads per Rsession for matrix math (openblas)
RhpcBLASctl::blas_set_num_threads(5)
# GENOMIC MATE SELECTION FUNCTIONS
source(here::here("code","gmsFunctions.R")) 
source(here::here("code","predCrossVar.R")) 

# GENOMIC RELATIONSHIP MATRICES (GRMS)
grms<-list(A=readRDS(file=here::here("output","kinship_A_IITA_2021May13.rds")),
           D=readRDS(file=here::here("output","kinship_domGenotypic_IITA_2021July5.rds")))

# DOSAGE MATRIX
dosages<-readRDS(file=here::here("data",
                                 "dosages_IITA_filtered_2021May13.rds"))

# BLUPs
blups<-readRDS(file=here::here("data","blups_forCrossVal.rds")) %>% 
  dplyr::select(-varcomp) %>% 
  rename(TrainingData=blups) # for compatibility with downstream functions

# SELECTION INDEX WEIGHTS
## from IYR+IK
## note that not ALL predicted traits are on index
SIwts<-c(logFYLD=20,
         HI=10,
         DM=15,
         MCMDS=-10,
         logRTNO=12,
         logDYLD=20,
         logTOPYLD=15,
         PLTHT=10) 
```

Run the "AD" and "DirDom" modelTypes built into `runGenomicPredictions()`. 

Output contains both GBLUPs for selection of indivuals _and_ SNP effects to use as input for prediction of cross usefulness and subsequent mate selection.

```{r run genomic predictions}
start<-proc.time()[3]
gpreds_ad<-runGenomicPredictions(modelType="AD",selInd=TRUE, SIwts=SIwts,
                                 getMarkEffs=TRUE,
                                 returnPEV=FALSE,
                                 blups=blups,grms=grms,dosages=dosages,
                                 ncores=20,nBLASthreads=5)
runtime<-proc.time()[3]-start; runtime/60
saveRDS(gpreds_ad,file = here::here("output","genomicPredictions_ModelAD.rds"))

gpreds_dirdom<-runGenomicPredictions(modelType="DirDom",selInd=TRUE, SIwts=SIwts,
                                     getMarkEffs=TRUE,
                                     returnPEV=FALSE,
                                     blups=blups,grms=grms,dosages=dosages,
                                     ncores=20,nBLASthreads=5)
runtime<-proc.time()[3]-start; runtime/60
saveRDS(gpreds_dirdom,file = here::here("output","genomicPredictions_ModelDirDom.rds"))

```

# Genomic mate predictions: cross usefulness

1. Use GBLUPs on SELIND to choose some to fraction of the clones.
2. For those selected parents, predict the SELIND usefulness for all pairwise matings.

```{bash, eval=F}
# 1) start a screen shell 
screen; # or screen -r if re-attaching...
# 2) start the singularity Linux shell inside that
singularity shell /workdir/$USER/rocker.sif; 
# Project directory, so R will use as working dir.
cd /home/mw489/implementGMSinCassava/;
# 3) Start R
R
```

Load input data
```{r load inputs for predictCrosses}
require(tidyverse); require(magrittr); 
# 5 threads per Rsession for matrix math (openblas)
RhpcBLASctl::blas_set_num_threads(5)
# GENOMIC MATE SELECTION FUNCTIONS
source(here::here("code","gmsFunctions.R")) 
source(here::here("code","predCrossVar.R")) 

# DOSAGE MATRIX
dosages<-readRDS(file=here::here("data",
                                 "dosages_IITA_filtered_2021May13.rds"))

# RECOMBINATION FREQUENCY MATRIX
recombFreqMat<-readRDS(file=here::here("data",
                                       "recombFreqMat_1minus2c_2021May13.rds"))
# HAPLOTYPE MATRIX
## keep only haplos for parents-in-the-pedigree
## those which will be used in prediction, saves memory
haploMat<-readRDS(file=here::here("data","haps_IITA_filtered_2021May13.rds"))
parents<-union(ped$sireID,ped$damID) 
parenthaps<-sort(c(paste0(parents,"_HapA"),
                   paste0(parents,"_HapB")))
haploMat<-haploMat[parenthaps,colnames(recombFreqMat)]

# SELECTION INDEX WEIGHTS
## from IYR+IK
## note that not ALL predicted traits are on index
SIwts<-c(logFYLD=20,
         HI=10,
         DM=15,
         MCMDS=-10,
         logRTNO=12,
         logDYLD=20,
         logTOPYLD=15,
         PLTHT=10) 

# SNP EFFECTS - FROM PREVIOUS STEP 
### Two models AD and DirDom
gpreds_ad<-readRDS(file = here::here("output","genomicPredictions_ModelAD.rds"))
gpreds_dirdom<-readRDS(file = here::here("output","genomicPredictions_ModelDirDom.rds"))
### Cor between SELIND GEBV and GETGV between models?
# left_join(gpreds_ad$gblups[[1]] %>% select(GID,predOf,SELIND) %>% rename(SELINDad=SELIND),
#           gpreds_dirdom$gblups[[1]] %>% select(GID,predOf,SELIND) %>% rename(SELINDdirdom=SELIND)) %>% 
#   group_by(predOf) %>% summarize(SELIND_corModels=cor(SELINDad,SELINDdirdom))
# predOf SELIND_corModels
# GEBV	0.9890091			
# GETGV	0.9373170	

# SELECTION OF CROSSES-TO-BE-PREDICTED
nParentsToSelect<-100
union_bestGEBVandGETGVdirdom<-union(gpreds_dirdom$gblups[[1]] %>% 
                                         filter(predOf=="GEBV") %>% 
                                         arrange(desc(SELIND)) %>% 
                                         slice(1:nParentsToSelect) %$% GID,
                                       gpreds_dirdom$gblups[[1]] %>% 
                                         filter(predOf=="GETGV") %>% 
                                         arrange(desc(SELIND)) %>% 
                                         slice(1:nParentsToSelect) %$% GID)
union_bestGEBVandGETGVad<-union(gpreds_ad$gblups[[1]] %>% 
                                         filter(predOf=="GEBV") %>% 
                                         arrange(desc(SELIND)) %>% 
                                         slice(1:nParentsToSelect) %$% GID,
                                       gpreds_ad$gblups[[1]] %>% 
                                         filter(predOf=="GETGV") %>% 
                                         arrange(desc(SELIND)) %>% 
                                         slice(1:nParentsToSelect) %$% GID)
union_bestGEBVandGETGV<-union(union_bestGEBVandGETGVdirdom,
                              union_bestGEBVandGETGVad)
length(union_bestGEBVandGETGV) 
# [1] 121 parents in top nParentsToSelect on SELIND for GEBV/GETGV - DirDom/AD model.

CrossesToPredict<-crosses2predict(union_bestGEBVandGETGV)
nrow(CrossesToPredict)
# [1] 7381 
```


```{r inputs for testing}
# snpeffs<-gpreds_dirdom$genomicPredOut[[1]] %>% slice(1:2)
# SIwts %<>% .[snpeffs$Trait]
# CrossesToPredict %<>% slice(1:2)
# modelType="DirDom"
# ncores=5
# selInd=TRUE
# stdSelInt = 2.67 # for computing UC; default equiv. to ~1% within-fam selection
```

```{r predictCrosses function}
predictCrosses<-function(modelType,
                         stdSelInt = 2.67, 
                         selInd,SIwts = NULL,
                         CrossesToPredict,
                         snpeffs,dosages,
                         haploMat,recombFreqMat,ncores){
  ## Format SNP effect matrices ########################
  
  AlleleSubEffectList<-snpeffs$allelesubsnpeff %>% 
    `names<-`(.,snpeffs$Trait) %>% 
    map(.,~t(.))
  
  if(modelType=="AD"){
    # DomDevEffects for model "AD" to predict VarTGV = VarBV + VarDD
    DomDevEffectList=snpeffs$domdevsnpeff %>% 
      `names<-`(.,snpeffs$Trait) %>% 
      map(.,~t(.)) }
  
  if(modelType=="DirDom"){
    # AddEffectList + DomEffectList --> VarTGV; AlleleSubEffectList --> VarBV;
    AddEffectList<-snpeffs$addsnpeff %>% 
      `names<-`(.,snpeffs$Trait) %>% 
      map(.,~t(.))
    DomEffectList<-snpeffs$domsnpeff %>% 
      `names<-`(.,snpeffs$Trait) %>% 
      map(.,~t(.)) }
  
  ## Predict cross variances ########################
  print("Predicting cross variance parameters")
  if(modelType=="A"){
    predvars<-predCrossVars(CrossesToPredict=CrossesToPredict,
                            AddEffectList=AlleleSubEffectList,
                            modelType="A",
                            haploMat=haploMat,
                            recombFreqMat=recombFreqMat,
                            ncores=ncores) %>%
      unnest(predVars) %>%
      mutate(predOf="VarBV") }
  if(modelType=="AD"){
    predvars<-predCrossVars(CrossesToPredict=CrossesToPredict,
                            AddEffectList=AlleleSubEffectList,
                            DomEffectList=DomDevEffectList,
                            modelType="AD",
                            haploMat=haploMat,
                            recombFreqMat=recombFreqMat,
                            ncores=ncores) %>% 
      unnest(predVars) %>%
      mutate(predOf=ifelse(predOf=="VarA","VarBV","VarDD"))
  }
  if(modelType=="DirDom"){
    predVarTGV<-predCrossVars(CrossesToPredict=CrossesToPredict,
                              AddEffectList=AddEffectList,
                              DomEffectList=DomEffectList,
                              modelType="AD", # no "DirDom" model in predCrossVars() nor is it needed
                              haploMat=haploMat,
                              recombFreqMat=recombFreqMat,
                              ncores=ncores)
    predVarBV<-predCrossVars(CrossesToPredict=CrossesToPredict,
                             AddEffectList=AlleleSubEffectList,
                             DomEffectList=NULL,
                             modelType="A", # no "DirDom" model in predCrossVars() nor is it needed
                             haploMat=haploMat,
                             recombFreqMat=recombFreqMat,
                             ncores=ncores)
    predvars<-predVarBV %>%
      unnest(predVars) %>%
      mutate(predOf="VarBV") %>%
      bind_rows(predVarTGV %>%
                  unnest(predVars)) }
  
  ## Predict cross means ########################
  print("Predicting cross means")
  ### predict MeanBVs
  predmeans<-predCrossMeans(AddEffectList=AlleleSubEffectList,
                            CrossesToPredict=CrossesToPredict,
                            doseMat=dosages,ncores=ncores,predType="BV")
  if(modelType=="AD"){ 
    ### DO NOT predict MeanTGV ~but~ duplicate MeanBV as MeanTGV prediction
    ### there IS predVarTGV for this model, output predUC-TGV (i.e. UC_variety)
    predmeans %<>%
      bind_rows(predmeans %>% mutate(predOf="TGV")) }
  
    predmeans
  if(modelType=="DirDom"){
    ### predict MeanTGVs
    ####  Prediction of MeanTGV is only available for the DirDom model
    #### or a model with "genotypic" additive-dominance SNP effects
    #### As implemented, modelType="AD" is the "classical" partition (BVs+ DomDevs)
    predmeans %<>%
      bind_rows(predCrossMeans(AddEffectList=AddEffectList,
                               DomEffectList=DomEffectList,
                               CrossesToPredict=CrossesToPredict,
                               doseMat=dosages,ncores=ncores,
                               predType="TGV"))
  }
  
  ## SIMPLIFIED, TIDY, CROSS-WISE OUTPUT ########################
  # store raw mean and var preds
  rawPreds<-list(predMeans=list(predmeans),
                 predVArs=list(predvars))
  
  ## tidy pred. means #####
  predmeans %<>% 
    mutate(predOf=gsub("Mean","",predOf),
           Trait2=Trait) %>% # to match with variance pred. output
    rename(Trait1=Trait) %>% # to match with variance pred. output
    select(sireID,damID,predOf,Trait1,Trait2,predMean)
  
  ## tidy pred. vars #####
  predvars %<>% 
    select(sireID,damID,Nsegsnps,predOf,Trait1,Trait2,predVar) %>% 
    mutate(predOf=gsub("Var","",predOf))
  if(modelType=="AD"){
    predvars %<>% 
      filter(predOf=="BV") %>% 
      bind_rows(predvars %>% 
                  pivot_wider(names_from = "predOf",
                              values_from = "predVar",
                              names_prefix = "predVar") %>% 
                  mutate(predVar=predVarBV+predVarDD,
                         predOf="TGV") %>% 
                  select(-predVarBV,-predVarDD))
  }
  if(modelType=="DirDom"){
    predvars %<>% 
      filter(predOf=="BV") %>% 
      bind_rows(predvars %>% 
                  filter(predOf!="BV") %>% 
                  pivot_wider(names_from = "predOf",
                              values_from = "predVar",
                              names_prefix = "predVar") %>% 
                  mutate(predVar=predVarA+predVarD,
                         predOf="TGV") %>% 
                  select(-predVarA,-predVarD))
  }
  
  ## SELECTION INDEX MEANS AND VARIANCES ########################
  #### Compute and add to tidy output, if requested
  if(selInd){
    print("Computing SELECTION INDEX means and variances.")
    traits<-unique(predmeans$Trait1)
    ## Compute Mean SELIND
    predmeans %<>% 
      select(-Trait2) %>% 
      spread(Trait1,predMean) %>% 
      select(sireID,damID,predOf,all_of(names(SIwts))) %>% 
      mutate(SELIND=as.numeric((predmeans %>% 
                                  select(-Trait2) %>% 
                                  spread(Trait1,predMean) %>%
                                  select(all_of(names(SIwts))) %>% 
                                  as.matrix(.))%*%SIwts)) %>%
      relocate(SELIND, .after = predOf) %>% 
      pivot_longer(cols = c(SELIND,all_of(traits)),
                   names_to = "Trait1",
                   values_to = "predMean") %>% 
      mutate(Trait2=Trait1) %>% 
      select(sireID,damID,predOf,Trait1,Trait2,predMean)
    ## Compute Var SELIND
    require(furrr); plan(multicore, workers = ncores)
    options(future.globals.maxSize=+Inf); options(future.rng.onMisuse="ignore")

    predvars %<>% 
      nest(predVars=c(Trait1,Trait2,predVar)) %>% 
      ## loop over each rep-fold-predOf-sireIDxdamID
      mutate(predVars=future_map(predVars,function(predVars){
        gmat<-predVars %>%
          pivot_wider(names_from = "Trait2",
                      values_from = "predVar") %>%
          column_to_rownames(var = "Trait1") %>%
          as.matrix
        gmat[lower.tri(gmat)]<-t(gmat)[lower.tri(gmat)]
        gmat %<>% .[names(SIwts),names(SIwts)]
        predSelIndVar<-SIwts%*%gmat%*%SIwts
        ## add sel index predictions to component trait
        ## var-covar predictions
        
        predVars<-tibble(Trait1="SELIND",Trait2="SELIND",
                         predVar=as.numeric(predSelIndVar)) %>%
          bind_rows(predVars)
        return(predVars) })) %>% 
      unnest(predVars)
  }
  
  ## USEFULNESS CRITERIA ########################
  tidyPreds<-predvars %>% 
    inner_join(predmeans) %>% 
    rename(Trait=Trait1) %>% 
    select(sireID,damID,Nsegsnps,predOf,Trait,predMean,predVar) %>% 
    mutate(predSD=sqrt(predVar),
           predUsefulness=predMean+(stdSelInt*predSD))
  
  predictions<-tibble(tidyPreds=list(tidyPreds),
                      rawPreds=list(rawPreds))
  return(predictions)
}
```

```{r test predictCrosses functions}
test_ad<-predictCrosses(modelType="AD",stdSelInt = 2.67, 
                        selInd=TRUE, SIwts=SIwts,
                        CrossesToPredict=CrossesToPredict,
                        snpeffs=gpreds_ad$genomicPredOut[[1]] %>% slice(1:2),
                        dosages=dosages,
                        haploMat=haploMat,recombFreqMat=recombFreqMat,
                        ncores=ncores)

test_dirdom<-predictCrosses(modelType="DirDom",stdSelInt = 2.67, 
                        selInd=TRUE, SIwts=SIwts,
                        CrossesToPredict=CrossesToPredict,
                        snpeffs=gpreds_dirdom$genomicPredOut[[1]] %>% slice(1:2),
                        dosages=dosages,
                        haploMat=haploMat,recombFreqMat=recombFreqMat,
                        ncores=ncores)
# > test_ad$tidyPreds[[1]]
# # A tibble: 12 x 9
#    sireID   damID   Nsegsnps predOf Trait predMean predVar predSD predUsefulness
#    <chr>    <chr>      <int> <chr>  <chr>    <dbl>   <dbl>  <dbl>          <dbl>
#  1 TMS15F1… TMS15F…     6808 BV     SELI…  -46.9   7.56e+1  8.70         -23.7  
#  2 TMS15F1… TMS15F…     6808 BV     MCMDS    0.131 1.07e-2  0.103          0.407
#  3 TMS15F1… TMS15F…     6808 BV     DM      -3.04  3.24e-1  0.569         -1.52 
#  4 TMS15F1… 2013_1…    13724 BV     SELI…  -40.5   9.78e+1  9.89         -14.1  
#  5 TMS15F1… 2013_1…    13724 BV     MCMDS    0.493 1.62e-2  0.127          0.832
#  6 TMS15F1… 2013_1…    13724 BV     DM      -2.37  4.20e-1  0.648         -0.641
#  7 TMS15F1… TMS15F…     6808 TGV    SELI…  -46.9   8.72e+1  9.34         -22.0  
#  8 TMS15F1… TMS15F…     6808 TGV    MCMDS    0.131 1.17e-2  0.108          0.420
#  9 TMS15F1… TMS15F…     6808 TGV    DM      -3.04  3.74e-1  0.612         -1.41 
# 10 TMS15F1… 2013_1…    13724 TGV    SELI…  -40.5   1.08e+2 10.4          -12.7  
# 11 TMS15F1… 2013_1…    13724 TGV    MCMDS    0.493 1.71e-2  0.131          0.842
# 12 TMS15F1… 2013_1…    13724 TGV    DM      -2.37  4.65e-1  0.682         -0.551

# > test_dirdom$tidyPreds[[1]]
# # A tibble: 12 x 9
#    sireID   damID   Nsegsnps predOf Trait predMean predVar predSD predUsefulness
#    <chr>    <chr>      <int> <chr>  <chr>    <dbl>   <dbl>  <dbl>          <dbl>
#  1 TMS15F1… TMS15F…     6808 BV     SELI… -6.24e+1 8.25e+1  9.08         -38.1  
#  2 TMS15F1… TMS15F…     6808 BV     MCMDS  2.21e-1 1.14e-2  0.107          0.506
#  3 TMS15F1… TMS15F…     6808 BV     DM    -4.01e+0 3.55e-1  0.596         -2.42 
#  4 TMS15F1… 2013_1…    13724 BV     SELI… -5.35e+1 1.07e+2 10.4          -25.8  
#  5 TMS15F1… 2013_1…    13724 BV     MCMDS  5.62e-1 1.68e-2  0.130          0.909
#  6 TMS15F1… 2013_1…    13724 BV     DM    -3.19e+0 4.62e-1  0.680         -1.38 
#  7 TMS15F1… TMS15F…     6808 TGV    SELI…  9.36e+1 8.01e+1  8.95         118.   
#  8 TMS15F1… TMS15F…     6808 TGV    MCMDS -1.74e-3 1.10e-2  0.105          0.279
#  9 TMS15F1… TMS15F…     6808 TGV    DM     6.24e+0 3.44e-1  0.586          7.81 
# 10 TMS15F1… 2013_1…    13724 TGV    SELI…  1.19e+2 1.03e+2 10.1          146.   
# 11 TMS15F1… 2013_1…    13724 TGV    MCMDS  2.15e-1 1.65e-2  0.129          0.558
# 12 TMS15F1… 2013_1…    13724 TGV    DM     8.10e+0 4.42e-1  0.665          9.88 
```

Add `predictCrosses()` to `code/gmsFunctions.R` and implement a full-run / prediction. 

Run the "AD" and "DirDom" modelTypes. 

```{r predict UC - top 121 parents - model DirDom}
start<-proc.time()[3]
crossPreds_dirdom<-predictCrosses(modelType="DirDom",stdSelInt = 2.67, 
                              selInd=TRUE, SIwts=SIwts,
                              CrossesToPredict=CrossesToPredict,
                              snpeffs=gpreds_dirdom$genomicPredOut[[1]],
                              dosages=dosages,
                              haploMat=haploMat,recombFreqMat=recombFreqMat,
                              ncores=ncores)
runtime<-proc.time()[3]-start; runtime/60
saveRDS(crossPreds_dirdom,file = here::here("output","genomicMatePredictions_top121parents_ModelDirDom.rds"))
```


```{r predict UC - top 121 parents - model AD}
start<-proc.time()[3]
crossPreds_ad<-predictCrosses(modelType="AD",stdSelInt = 2.67, 
                              selInd=TRUE, SIwts=SIwts,
                              CrossesToPredict=CrossesToPredict,
                              snpeffs=gpreds_ad$genomicPredOut[[1]],
                              dosages=dosages,
                              haploMat=haploMat,recombFreqMat=recombFreqMat,
                              ncores=ncores)
runtime<-proc.time()[3]-start; runtime/60
saveRDS(crossPreds_ad,file = here::here("output","genomicMatePredictions_top121parents_ModelAD.rds"))
```

# [TO DO] Write predictions to disk

Add genetic groups and tidy format

```{r}
# unique_gids<-predModelA %>%
#   dplyr::select(genomicPredOut) %>%
#   unnest(genomicPredOut) %>%
#   select(-varcomps) %>%
#   unnest(gblups) %$%
#   GID %>%
#   unique
# 
# c1a<-unique_gids %>%
#   grep("c1a",.,value = T,ignore.case = T) %>%
#   union(.,unique_gids %>%
#           grep("^F",.,value = T,ignore.case = T) %>%
#           grep("c1b",.,value = T,ignore.case = T,invert = T))
# c1b<-unique_gids%>% grep("c1b",.,value = T,ignore.case = T)
# c2a<-unique_gids %>%
#   grep("C2a",.,value = T,ignore.case = T) %>%
#   grep("NR17",.,value = T,ignore.case = T)
# c2b<-unique_gids %>%
#   grep("C2b",.,value = T,ignore.case = T) %>%
#   .[!. %in% c(c1a,c1b,c2a)]
# c3a<-unique_gids %>%
#   grep("C3a",.,value = T,ignore.case = T) %>%
#   .[!. %in% c(c1a,c1b,c2a,c2b)]
# c3b<-unique_gids %>%
#   grep("NR20C3",.,value = T,ignore.case = T) %>%
#   .[!. %in% c(c1a,c1b,c2a,c2b,c3a)]
# nrTP<-setdiff(unique_gids,unique(c(c1a,c1b,c2a,c2b,c3a,c3b)))
```

Write CSV's

```{r}
# ## Format and write GEBV
# predModelA %>% 
#   select(Trait,genomicPredOut) %>% 
#   unnest(genomicPredOut) %>% 
#   select(-varcomps) %>% 
#   unnest(gblups) %>% 
#   select(-GETGV,-contains("PEV")) %>%
#   spread(Trait,GEBV) %>% 
#   mutate(Group=case_when(GID %in% nrTP ~ "nrTP",
#                          GID %in% c1a ~ "C1a",
#                          GID %in% c1b ~ "C1b",
#                          GID %in% c2a ~ "C2a",
#                          GID %in% c2b ~ "C2b",
#                          GID %in% c3a ~ "C3a",
#                          GID %in% c3b ~ "C3b")) %>% 
#   select(Group,GID,any_of(traits)) %>% 
#   arrange(desc(Group)) %>% 
#   write.csv(., file = here::here("output","GEBV_NRCRI_ModelA_2021May03.csv"), row.names = F)
# 
# ## Format and write GETGV
# predModelADE %>% 
#   select(Trait,genomicPredOut) %>% 
#   unnest(genomicPredOut) %>% 
#   select(-varcomps) %>% 
#   unnest(gblups) %>% 
#   select(GID,Trait,GETGV) %>% 
#   spread(Trait,GETGV) %>% 
#   mutate(Group=case_when(GID %in% nrTP ~ "nrTP",
#                          GID %in% c1a ~ "C1a",
#                          GID %in% c1b ~ "C1b",
#                          GID %in% c2a ~ "C2a",
#                          GID %in% c2b ~ "C2b",
#                          GID %in% c3a ~ "C3a",
#                          GID %in% c3b ~ "C3b")) %>% 
#   select(Group,GID,any_of(traits)) %>% 
#   arrange(desc(Group)) %>% 
#   write.csv(., file = here::here("output","GETGV_NRCRI_ModelADE_2021May03.csv"), row.names = F)
# 
# ### Make a unified "tidy" long-form: 
# predModelA %>% 
#   select(Trait,genomicPredOut) %>% 
#   unnest(genomicPredOut) %>% 
#   select(-varcomps) %>% 
#   unnest(gblups) %>% 
#   select(-GETGV) %>% 
#   full_join(predModelADE %>% 
#               select(Trait,genomicPredOut) %>% 
#               unnest(genomicPredOut) %>% 
#               select(-varcomps) %>% 
#               unnest(gblups) %>% 
#               rename(GEBV_modelADE=GEBV,
#                      PEV_modelADE=PEVa) %>% 
#               select(-genomicPredOut)) %>% 
#   mutate(Group=case_when(GID %in% nrTP ~ "nrTP",
#                          GID %in% c1a ~ "C1a",
#                          GID %in% c1b ~ "C1b",
#                          GID %in% c2a ~ "C2a",
#                          GID %in% c2b ~ "C2b",
#                          GID %in% c3a ~ "C3a",
#                          GID %in% c3b ~ "C3b")) %>% 
#   relocate(Group,.before = GID) %>% 
#   write.csv(., file = here::here("output","genomicPredictions_NRCRI_2021May03.csv"), row.names = F)
```




# Results

See [Results](07-Results.html): Home for plots and summary tables.
