---
title: "Results"
site: workflowr::wflow_site
date: "2021-June-07"
output: 
  workflowr::wflow_html:
    code_folding: hide
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = F, 
                      tidy='styler', tidy.opts=list(strict=FALSE,width.cutoff=100), highlight=TRUE)
```

# Raw data

Summary of the number of unique plots, locations, years, etc. in the cleaned plot-basis data. See [here](01-cleanTPdata.html) for details..

```{r}
library(tidyverse); library(magrittr); library(ragg)
rawdata<-readRDS(file=here::here("output","IITA_ExptDesignsDetected_2021May10.rds"))
rawdata %>% 
  summarise(Nplots=nrow(.),
            across(c(locationName,studyYear,studyName,TrialType,GID), ~length(unique(.)),.names = "N_{.col}")) %>% 
  rmarkdown::paged_table()
```
This _is not_ the same number of clones as are expected to be genotyped-and-phenotyped.

Break down the plots based on the trial design and TrialType (really a grouping of the population that is breeding program specific), captured by two logical variables, **CompleteBlocks** and **IncompleteBlocks**.  
```{r}
rawdata %>% 
  count(TrialType,CompleteBlocks,IncompleteBlocks) %>% 
  spread(TrialType,n) %>% 
  rmarkdown::paged_table()
```

Next, look at breakdown of plots by TrialType (rows) and locations (columns):
```{r, cols.print=16}
rawdata %>% 
  count(locationName,TrialType) %>% 
  spread(locationName,n) %>% 
  rmarkdown::paged_table()
```

```{r}
traits<-c("MCMDS","DM","PLTHT","BRNHT1","BRLVLS","HI",
          "logDYLD", # <-- logDYLD now included. 
          "logFYLD","logTOPYLD","logRTNO","TCHART","LCHROMO","ACHROMO","BCHROMO")
rawdata %>% 
  select(locationName,studyYear,studyName,TrialType,any_of(traits)) %>% 
  pivot_longer(cols = any_of(traits), values_to = "Value", names_to = "Trait") %>% 
  ggplot(.,aes(x=Value,fill=Trait)) + geom_histogram() + facet_wrap(~Trait, scales='free') + 
  theme_bw() + scale_fill_viridis_d() + 
  labs(title = "Distribution of Raw Phenotypic Values")
```
How many genotyped-and-phenotyped clones?

```{r}
rawdata %>% 
  select(locationName,studyYear,studyName,TrialType,germplasmName,FullSampleName,GID,any_of(traits)) %>% 
  pivot_longer(cols = any_of(traits), values_to = "Value", names_to = "Trait") %>%
  filter(!is.na(Value),!is.na(FullSampleName)) %>%
  distinct(germplasmName,FullSampleName,GID) %>% 
  rmarkdown::paged_table()
```
There are 8149 genotyped-and-phenotyped clones! 

# BLUPs

These are the BLUPs combining data for each clone across trials/locations _without_ genomic information, used as input for genomic prediction downstream.

```{r, message=F, rows.print=12}
blups<-readRDS(file=here::here("data","blups_forCrossVal.rds")) 
gidWithBLUPs<-blups %>% select(Trait,blups) %>% unnest(blups) %$% unique(GID)
rawdata %>% 
  select(observationUnitDbId,GID,any_of(blups$Trait)) %>% 
  pivot_longer(cols = any_of(blups$Trait), 
               names_to = "Trait", 
               values_to = "Value",values_drop_na = T) %>% 
  filter(GID %in% gidWithBLUPs) %>% 
  group_by(Trait) %>% 
  summarize(Nplots=n()) %>% 
  ungroup() %>% 
  left_join(blups %>% 
              mutate(Nclones=map_dbl(blups,~nrow(.)),
                     avgREL=map_dbl(blups,~mean(.$REL)),
                     Vg=map_dbl(varcomp,~.["GID!GID.var","component"]),
                     Ve=map_dbl(varcomp,~.["R!variance","component"]),
                     H2=Vg/(Vg+Ve)) %>% 
              select(-blups,-varcomp)) %>% 
  mutate(across(is.numeric,~round(.,3))) %>% arrange(desc(H2)) %>% 
  rmarkdown::paged_table()
```

```{r}
blups %>% 
  select(Trait,blups) %>% 
  unnest(blups) %>% 
  ggplot(.,aes(x=drgBLUP,fill=Trait)) + geom_histogram() + facet_wrap(~Trait, scales='free') + 
  theme_bw() + scale_fill_viridis_d() + theme(legend.position = 'none') + 
  labs(title = "Distribution of de-regressed BLUP Values")
```

```{r}
blups %>% 
  select(Trait,blups) %>% 
  unnest(blups) %>% 
  ggplot(.,aes(x=Trait,y=REL,fill=Trait)) + geom_boxplot(notch=T) + #facet_wrap(~Trait, scales='free') + 
  theme_bw() + scale_fill_viridis_d() + 
  theme(axis.text.x = element_text(angle=90),
        legend.position = 'none') + 
  labs(title = "Distribution of BLUP Reliabilities")
```
# Marker density and distribution
```{r}
library(tidyverse); library(magrittr);
snps<-readRDS(file=here::here("data","dosages_IITA_filtered_2021May13.rds"))
mrks<-colnames(snps) %>% 
  tibble(SNP_ID=.) %>% 
  separate(SNP_ID,c("Chr","Pos","Allele"),"_") %>% 
  mutate(Chr=as.integer(gsub("S","",Chr)),
         Pos=as.numeric(Pos))
mrks %>% 
  ggplot(.,aes(x=Pos,fill=as.character(Chr))) + geom_histogram() + 
  facet_wrap(~Chr,scales = 'free') + theme_bw() + 
  scale_fill_viridis_d() + theme(legend.position = 'none',
                                 axis.text.x = element_text(angle=90))
```
```{r, rows.print=18}
mrks %>% count(Chr) %>% rmarkdown::paged_table()
```

# Pedigree

Summarize the pedigree and the [verification results described here](03-validatePedigree.html).

```{r}
library(tidyverse); library(magrittr);
ped2check_genome<-readRDS(file=here::here("output","ped2check_genome.rds"))
ped2check_genome %<>% 
  select(IID1,IID2,Z0,Z1,Z2,PI_HAT)
ped2check<-read.table(file=here::here("output","ped2genos.txt"),
                      header = F, stringsAsFactors = F) %>% 
  rename(FullSampleName=V1,DamID=V2,SireID=V3)
ped2check %<>% 
  select(FullSampleName,DamID,SireID) %>% 
  inner_join(ped2check_genome %>% 
               rename(FullSampleName=IID1,DamID=IID2) %>% 
               bind_rows(ped2check_genome %>% 
                           rename(FullSampleName=IID2,DamID=IID1))) %>% 
  distinct %>% 
  mutate(FemaleParent=case_when(Z0<0.32 & Z1>0.67~"Confirm",
                                       SireID==DamID & PI_HAT>0.6 & Z0<0.3 & Z2>0.32~"Confirm",
                                       TRUE~"Reject")) %>% 
  select(-Z0,-Z1,-Z2,-PI_HAT) %>% 
  inner_join(ped2check_genome %>% 
               rename(FullSampleName=IID1,SireID=IID2) %>% 
               bind_rows(ped2check_genome %>% 
                           rename(FullSampleName=IID2,SireID=IID1))) %>% 
  distinct %>% 
  mutate(MaleParent=case_when(Z0<0.32 & Z1>0.67~"Confirm",
                                       SireID==DamID & PI_HAT>0.6 & Z0<0.3 & Z2>0.32~"Confirm",
                                       TRUE~"Reject")) %>% 
  select(-Z0,-Z1,-Z2,-PI_HAT)
rm(ped2check_genome)

ped2check %<>% 
  mutate(Cohort=NA,
         Cohort=ifelse(grepl("TMS18",FullSampleName,ignore.case = T),"TMS18",
                       ifelse(grepl("TMS15",FullSampleName,ignore.case = T),"TMS15",
                              ifelse(grepl("TMS14",FullSampleName,ignore.case = T),"TMS14",
                                     ifelse(grepl("TMS13|2013_",FullSampleName,ignore.case = T),"TMS13","GGetc")))))
```

Proportion of accessions with male, female or both parents in pedigree confirm-vs-rejected?
```{r}
ped2check %>% 
  count(FemaleParent,MaleParent) %>% 
  mutate(Prop=round(n/sum(n),2))
```

Proportion of accessions _within_ each Cohort with pedigree records confirmed-vs-rejected?
```{r}
ped2check %>% 
  count(Cohort,FemaleParent,MaleParent) %>% 
  spread(Cohort,n) %>% 
  mutate(across(is.numeric,~round(./sum(.),2))) %>% 
  rmarkdown::paged_table()
```

Use only fully-confirmed families / trios. 
Remove any without both parents confirmed.


```{r}
ped<-read.table(here::here("output","verified_ped.txt"),
                header = T, stringsAsFactors = F) %>% 
  mutate(Cohort=NA,
         Cohort=ifelse(grepl("TMS18",FullSampleName,ignore.case = T),"TMS18",
                       ifelse(grepl("TMS15",FullSampleName,ignore.case = T),"TMS15",
                              ifelse(grepl("TMS14",FullSampleName,ignore.case = T),"TMS14",
                                     ifelse(grepl("TMS13|2013_",FullSampleName,ignore.case = T),"TMS13","GGetc")))))
```
Summary of family sizes
```{r}
ped %>% 
  count(SireID,DamID) %$% summary(n)
```
```{r}
ped %>% nrow(.) # 4259 pedigree entries
ped %>% 
  count(Cohort,name = "Number of Verified Pedigree Entries")
```
```{r}
ped %>% 
  distinct(Cohort,SireID,DamID) %>% 
  count(Cohort,name = "Number of Families per Cohort")
```
730 families. Mean size 5.85, range 1-77.

# Parent-wise Cross-validation

```{r}
parentfolds<-readRDS(file=here::here("output","parentfolds.rds"))
summarized_parentfolds<-parentfolds %>% 
  mutate(Ntestparents=map_dbl(testparents,length),
         Ntrainset=map_dbl(trainset,length),
         Ntestset=map_dbl(testset,length),
         NcrossesToPredict=map_dbl(CrossesToPredict,nrow)) %>% 
  select(Repeat,Fold,starts_with("N"))
summarized_parentfolds %>% 
  rmarkdown::paged_table()
```
```{r}
summarized_parentfolds %>% summarize(across(is.numeric,median,.names = "median{.col}"))
```

Selection index weights
```{r}
c(logFYLD=20,
  HI=10,
  DM=15,
  MCMDS=-10,
  logRTNO=12,
  logDYLD=20,
  logTOPYLD=15,
  PLTHT=10)
```


## Prediction accuracy

3. [Check prediction accuracy](03-CrossValidation.html): Evaluate prediction accuracy with cross-validation.

### Selection Index Accuracy
Selection index weights
```{r}
# SELECTION INDEX WEIGHTS
## from IYR+IK
## note that not ALL predicted traits are on index
SIwts<-c(logFYLD=20,
         HI=10,
         DM=15,
         MCMDS=-10,
         logRTNO=12,
         logDYLD=20,
         logTOPYLD=15,
         PLTHT=10) 
SIwts
```

```{r}
library(ggdist)
accuracy_ad<-readRDS(here::here("output","cvAD_5rep5fold_predAccuracy.rds"))
accuracy_dirdom<-readRDS(here::here("output","cvDirDom_5rep5fold_predAccuracy.rds"))
accuracy<-accuracy_ad$meanPredAccuracy %>% 
  bind_rows(accuracy_dirdom$meanPredAccuracy) %>% 
  filter(Trait=="SELIND") %>% 
  mutate(VarComp=gsub("Mean","",predOf),
         predOf="Mean") %>% 
  bind_rows(accuracy_ad$varPredAccuracy %>% 
              bind_rows(accuracy_dirdom$varPredAccuracy) %>% 
              filter(Trait1=="SELIND") %>% 
              rename(Trait=Trait1) %>% 
              select(-Trait2) %>% 
              mutate(VarComp=gsub("Var","",predOf),
                     predOf="Var")) %>% 
  select(-predVSobs)

colors<-viridis::viridis(4)[1:2]
```

```{r}
accuracy %>% 
  mutate(predOf=factor(predOf,levels=c("Mean","Var")),
         VarComp=factor(VarComp,levels=c("BV","TGV"))) %>% 
  ggplot(.,aes(x=VarComp,y=AccuracyEst,fill=VarComp)) +
  ggdist::stat_halfeye(adjust=0.5,.width = 0,fill='gray',width=0.75) +  
  geom_boxplot(width=0.12,notch = TRUE) +
  ggdist::stat_dots(side = "left",justification = 1.1,
                    binwidth = 0.03,dotsize=0.6) +
  theme_bw() + 
  scale_fill_manual(values = colors) + 
  geom_hline(yintercept = 0, color='black', size=0.9) +
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        strip.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.title = element_text(face='bold',color = 'black'),
        strip.text.x = element_text(face='bold',color='black',size=14),
        axis.text.y = element_text(face = 'bold',color='black'),
        legend.text = element_text(face='bold'),
        legend.position = 'bottom') +
  facet_grid(predOf~modelType,scales = 'free') + 
  #facet_wrap(~predOf+modelType,scales = 'free_y',nrow=1) + 
  labs(title="Selection Index Prediction Accuracy") + 
  coord_cartesian(xlim = c(1.2, NA))
```


### Accuracy predicting cross-means
```{r,fig.width = 10, fig.height=6}
accuracy_ad$meanPredAccuracy %>% 
  bind_rows(accuracy_dirdom$meanPredAccuracy) %>% 
  select(-predVSobs) %>% 
  mutate(Trait=factor(Trait,levels=c("SELIND",blups$Trait)),
         predOf=factor(paste0(predOf,"_",modelType),levels=c("MeanBV_AD","MeanTGV_AD","MeanBV_DirDom","MeanTGV_DirDom"))) %>% 
  ggplot(.,aes(x=predOf,y=AccuracyEst,fill=predOf,color=modelType)) + 
  geom_boxplot(width=0.4,notch = TRUE, color='gray40') +
  ggdist::stat_dots(side = "left", justification = 1.3,
                    binwidth = 0.03,dotsize=0.5,layout="swarm") +
  scale_fill_manual(values = viridis::viridis(4)[1:4]) + 
  scale_color_manual(values = viridis::viridis(4)[1:4]) + 
  geom_hline(yintercept = 0, color='black', size=0.8) +
  facet_grid(.~Trait) + 
  labs(title="Accuracy predicting cross-means") + 
  theme_bw() + 
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        strip.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.text.y = element_text(face='bold'),
        strip.text.x = element_text(face='bold'),
        legend.position='bottom',
        legend.text = element_text(face='bold'),
        panel.spacing = unit(0.05, "lines"))
```

### Accuracy predicting variance params
```{r,fig.width = 10, fig.height=5}
accuracy_ad$varPredAccuracy %>% 
  bind_rows(accuracy_dirdom$varPredAccuracy) %>% 
  select(-predVSobs) %>% 
  filter(Trait1==Trait2) %>% 
  mutate(Trait1=factor(Trait1,levels=c("SELIND",blups$Trait)),
         predOf=factor(paste0(predOf,"_",modelType),
                       levels=c("VarBV_AD","VarTGV_AD",
                                "VarBV_DirDom","VarTGV_DirDom"))) %>% 
  ggplot(.,aes(x=predOf,y=AccuracyEst,fill=predOf,color=modelType)) + 
  geom_boxplot(width=0.4,notch = TRUE,color='gray40') +
  ggdist::stat_dots(side = "left", justification = 1.3,layout='swarm',
                    binwidth = 0.03,dotsize=0.4) +
  scale_fill_manual(values = viridis::viridis(4)[1:4]) + 
  scale_color_manual(values = viridis::viridis(4)[1:4]) + 
  geom_hline(yintercept = 0, color='black', size=0.8) +
  facet_grid(.~Trait1) + 
  labs(title="Accuracy predicting cross-variances") + 
  theme_bw() + 
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        strip.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.text.y = element_text(face='bold'),
        strip.text.x = element_text(face='bold'),
        legend.position='bottom',
        legend.text = element_text(face='bold'),
        panel.spacing = unit(0.05, "lines"))
```


```{r, fig.width = 9, fig.height = 9}
accuracy_ad$varPredAccuracy %>% 
  bind_rows(accuracy_dirdom$varPredAccuracy) %>% 
  select(-predVSobs) %>% 
  filter(Trait1!="SELIND",Trait2!="SELIND",
         Trait1!=Trait2) %>% 
  mutate(#Trait1=factor(Trait1,levels=c("SELIND",blups$Trait)),
         #Trait2=factor(Trait2,levels=c("SELIND",blups$Trait)),
         Trait1=factor(Trait1,levels=c(blups$Trait)),
         Trait2=factor(Trait2,levels=c(blups$Trait)),
         predOf=factor(paste0(predOf,"_",modelType),
                       levels=c("VarBV_AD","VarTGV_AD",
                                "VarBV_DirDom","VarTGV_DirDom"))) %>% 
  ggplot(.,aes(x=predOf,y=AccuracyEst,fill=predOf,color=modelType)) + 
  geom_boxplot(notch = TRUE) +
  scale_fill_manual(values = viridis::viridis(4)[1:4]) + 
  scale_color_manual(values = viridis::viridis(4)[1:4]) + 
  geom_hline(yintercept = 0, color='gray40', size=0.6) +
  facet_wrap(~Trait1+Trait2,nrow=5) + 
  labs(title="Accuracy predicting cross-covariances") + 
  theme_bw() + 
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        strip.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.text.y = element_text(face='bold'),
        strip.text.x = element_text(size=8),
        strip.text.y = element_text(size=8,angle = 0),
        legend.position='bottom',
        legend.text = element_text(face='bold'),
        panel.spacing = unit(0.05, "lines"))
```

## [TO REDO] Accuracy based only on large families

What if we only compute accuracy with families having >10 or >20 members? 
```{r, eval=F}
# library(tidyverse); library(magrittr)
# cvvars<-readRDS(here::here("output","cvVarPredAccuracyAD.rds"))
# 
# big_acc<-cvvars %>% 
#   mutate(AccuracyEst_n10=map_dbl(predVSobs,function(predVSobs){
#     z<-predVSobs %>% filter(famSize>=10)
#     out<-psych::cor.wt(z[,c("predVar","obsVar")],
#                        w = z$famSize) %$% r[1,2] %>%
#       round(.,3)
#     return(out) }),
#     AccuracyEst_n20=map_dbl(predVSobs,function(predVSobs){
#       z<-predVSobs %>% filter(famSize>=20)
#       out<-psych::cor.wt(z[,c("predVar","obsVar")],
#                          w = z$famSize) %$% r[1,2] %>%
#         round(.,3)
#       return(out) }))
# 
# big_acc %<>% 
#   mutate(Trait1=factor(Trait1,levels=c("SELIND",blups$Trait)),
#          Trait2=factor(Trait2,levels=c("SELIND",blups$Trait)),
#          predOf=factor(predOf,levels=c("VarBV","VarTGV")))
# 
# big_acc %>% 
#   filter(Trait1==Trait2,predOf=="VarBV") %>%
#   select(-predVSobs) %>% 
#   pivot_longer(cols = contains("Accuracy"), names_to = "FamilySizeForValidation",values_to = "AccuracyEst") %>% 
#   ggplot(.,aes(x=predOf,y=AccuracyEst,fill=FamilySizeForValidation, color=predOf)) + 
#   geom_boxplot(width=0.4,notch = TRUE,color='gray40') +
#   scale_fill_manual(values = viridis::viridis(4)[1:3]) + 
#   #scale_color_manual(values = viridis::viridis(4)[1:3]) + 
#   geom_hline(yintercept = 0, color='black', size=0.8) +
#   facet_grid(predOf~Trait1) + 
#   labs(title="Accuracy predicting cross-variances in bigger famiilies only?") + 
#   theme_bw() + 
#   theme(axis.text.x = element_blank(),
#         axis.title.x = element_blank(),
#         strip.background = element_blank(),
#         panel.grid.major = element_blank(),
#         panel.grid.minor = element_blank(),
#         axis.text.y = element_text(face='bold'),
#         strip.text.x = element_text(face='bold'),
#         legend.position='bottom',
#         legend.text = element_text(face='bold'),
#         panel.spacing = unit(0.05, "lines"))
```

# [RUNNING] Genomic Mate Selection

Placeholder for summaries of the full-model predictions of cross means and variances to-be-used for selection.

# Genetic Gain Estimates

```{r}
# GBLUPs
### Two models AD and DirDom
#gpreds_ad<-readRDS(file = here::here("output","genomicPredictions_ModelAD.rds"))
gpreds_dirdom<-readRDS(file = here::here("output","genomicPredictions_ModelDirDom.rds"))
si_getgvs<-gpreds_dirdom$gblups[[1]] %>% 
#si_getgvs<-gpreds_ad$gblups[[1]] %>% 
  filter(predOf=="GETGV") %>% 
  select(GID,SELIND)

## IITA Germplasm Ages
ggcycletime<-readxl::read_xls(here::here("data","PedigreeGeneticGainCycleTime_aafolabi_01122020.xls")) %>% 
  mutate(Year_Accession=as.numeric(Year_Accession))

# Need germplasmName field from raw trial data to match GEBV and cycle time
rawdata<-readRDS(here::here("output","IITA_ExptDesignsDetected_2021May10.rds"))
si_getgvs %<>% 
  left_join(rawdata %>% 
              distinct(germplasmName,GID)) %>% 
  group_by(GID) %>% 
  slice(1) %>% 
  ungroup()
# table(ggcycletime$Accession %in% si_getgvs$germplasmName)
# FALSE  TRUE 
#   193   614 

si_getgvs %<>% 
  left_join(.,ggcycletime %>% 
              rename(germplasmName=Accession) %>% 
              mutate(Year_Accession=as.numeric(Year_Accession))) %>% 
  mutate(Year_Accession=case_when(grepl("2013_|TMS13",germplasmName)~2013,
                                  grepl("TMS14",germplasmName)~2014,
                                  grepl("TMS15",germplasmName)~2015,
                                  grepl("TMS18",germplasmName)~2018,
                                  !grepl("2013_|TMS13|TMS14|TMS15|TMS18",germplasmName)~Year_Accession))

# Declare the "eras" as PreGS\<2012 and GS\>=2013.
si_getgvs %<>% 
  filter(Year_Accession>2012 | Year_Accession<2005)
si_getgvs %<>% 
  mutate(GeneticGroup=ifelse(Year_Accession>=2013,"GS","PreGS"))
```
Show rate of genetic gain - getgv vs. accession age
```{r}
# , fig.height=10, fig.width=12
si_getgvs %>% 
  select(GeneticGroup,GID,Year_Accession,SELIND) %>% 
  ggplot(.,aes(x=Year_Accession,y=SELIND,color=GeneticGroup)) + 
  geom_point(size=1.25) + geom_smooth(method=lm, se=TRUE, size=1.5) + 
  # facet_wrap(~Trait,scales='free_y', ncol=2) + 
  theme_bw() +
  theme(axis.text = element_text(face = 'bold',angle = 0, size=14),
        axis.title = element_text(face = 'bold',size=16),
        strip.background.x = element_blank(),
        strip.text = element_text(face='bold',size=18)) + 
  scale_color_viridis_d() + 
  labs(title = "Selection Index GETGV vs. Accession Age by 'era' [GS vs. PreGS]",
       subtitle = "SI GETGV from modelType='DirDom'")
```
Barplot of mean GEBV vs. Cycle
```{r}
si_gebvs<-gpreds_dirdom$gblups[[1]] %>% 
  filter(predOf=="GEBV") %>% 
  select(GID,SELIND) %>% 
  mutate(GeneticGroup=case_when(grepl("2013_|TMS13",GID)~"C1",
                                  grepl("TMS14",GID)~"C2",
                                  grepl("TMS15",GID)~"C3",
                                  grepl("TMS18",GID)~"C4",
                                  !grepl("2013_|TMS13|TMS14|TMS15|TMS18",GID)~"PreGS"),
         GeneticGroup=factor(GeneticGroup,levels = c("PreGS","C1","C2","C3","C4")))
si_gebvs %>% 
  group_by(GeneticGroup) %>% 
  summarize(meanGEBV=mean(SELIND),
            stdErr=sd(SELIND)/sqrt(n()),
            upperSE=meanGEBV+stdErr,
            lowerSE=meanGEBV-stdErr) %>% 
  ggplot(.,aes(x=GeneticGroup,y=meanGEBV,fill=GeneticGroup)) + 
  geom_bar(stat = 'identity', color='gray60', size=1.25) + 
  geom_linerange(aes(ymax=upperSE,
                     ymin=lowerSE), color='gray60', size=1.25) + 
  #facet_wrap(~Trait,scales='free_y', ncol=1) + 
  theme_bw() +
  geom_hline(yintercept = 0, size=1.15, color='black') + 
  theme(axis.text.x = element_text(face = 'bold',angle = 0, size=12),
        axis.title.y = element_text(face = 'bold',size=14),
        legend.position = 'none',
        strip.background.x = element_blank(),
        strip.text = element_text(face='bold',size=14)) + 
  scale_fill_viridis_d() + 
  labs(x=NULL,y="Mean GEBVs",
       title="Mean +/- Std. Error Selection Index GEBV by Cycle")
```



